<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Calculates Tryptase Variant Likelihoods • TrypLik</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Calculates Tryptase Variant Likelihoods">
<meta name="description" content="Calculates tryptase variant (alpha, beta) likelihoods based on counts using a wrapped C method. Preprocessing of cram files is provided as both bash scripts and R wrappers. A detailed description of the method employed is provided as a vignette. Contact HIBio to learn more about our tryptase program and services provided.">
<meta property="og:description" content="Calculates tryptase variant (alpha, beta) likelihoods based on counts using a wrapped C method. Preprocessing of cram files is provided as both bash scripts and R wrappers. A detailed description of the method employed is provided as a vignette. Contact HIBio to learn more about our tryptase program and services provided.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">TrypLik</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/TrypLik.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="tryplik">TrypLik<a class="anchor" aria-label="anchor" href="#tryplik"></a>
</h1></div>
<!-- badges: start -->

<p>The goal of <code>TrypLik</code> is to support tryptase copy number estimation from sequence data.</p>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/" class="external-link">CRAN</a>.</p>
<!-- Then install `TrypLik` from [Bioconductor](http://bioconductor.org/) using the following code: -->
<!-- ```{r 'install', eval = FALSE} -->
<!-- if (!requireNamespace("BiocManager", quietly = TRUE)) { -->
<!--     install.packages("BiocManager") -->
<!-- } -->
<!-- BiocManager::install("TrypLik") -->
<!-- ``` -->
<p>Install the development version from <a href="https://github.com/HIBio/TrypLik" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"HIBio/TrypLik"</span><span class="op">)</span></span></code></pre></div>
<p>or, if not using the Bioconductor ecosystem:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"HIBio/TrypLik"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h2>
<p>Provided the sequencing data has been processed (see the background vignette included in this package, and accompanying shell scripts in <code>inst/</code>) to produce the reads mapped to specific tryptase alleles, these can be provided to the <code>TrypLik</code> function</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://hibio.github.io/TrypLik/">"TrypLik"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="reference/TrypLik.html">TrypLik</a></span><span class="op">(</span><span class="fl">26</span>, <span class="fl">0</span>, <span class="fl">347</span>, <span class="fl">316</span>, <span class="fl">304</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Alpha_count Beta_count Beta_FS_count Posterior_likelihood</span></span>
<span><span class="co">#&gt; 1           2          2             0              0.99890</span></span>
<span><span class="co">#&gt; 2           2          3             0              0.00008</span></span>
<span><span class="co">#&gt; 3           3          2             0              0.00082</span></span>
<span><span class="co">#&gt; 4           3          3             0              0.00020</span></span></code></pre></div>
<p>This function is an R wrapper for the C function, producing a native <code>data.frame</code>.</p>
<p>The C code is provided as a standalone source file in <a href="inst/tryplik.c"><code>inst/tryplik.c</code></a> in which the R-related code has been commented out. The R wrapper uses the file <a href="src/tryplik.c"><code>src/tryplik.c</code></a> which only differs in the additional R components.</p>
<p>Once compiled, the standalone version can be used in a command-line pipeline; following the processing outlined in the background vignette, one could use</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">[...]</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">samtools</span> markdup <span class="at">-r</span> positionsort.sam temp.sam</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">./count_tryptase</span> <span class="op">&gt;&gt;</span> tryptase.out</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="fu">cat</span> tryptase.out <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'{printf "./Tryplik %d %d %d %d %d | sort -nk2 | tail -n 1\n", $6, $7, $8, $9, $10}'</span></span></code></pre></div>
<p>The processing can be entirely performed from R, provided that both <code>samtools</code> and <code>bwa</code> are installed. To preprocess the file <code>HG00100.final.cram</code> from 1000 Genomes, producing a <code>.sam</code> file suitable for counting</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="st">"HG00100.final.cram"</span><span class="op">)</span></span>
<span><span class="co"># Processing... this may take some time</span></span>
<span><span class="co"># [bwa_index] Pack FASTA... 0.00 sec</span></span>
<span><span class="co"># [bwa_index] Construct BWT for the packed sequence...</span></span>
<span><span class="co"># [bwa_index] 0.00 seconds elapse.</span></span>
<span><span class="co"># [bwa_index] Update BWT... 0.00 sec</span></span>
<span><span class="co"># [bwa_index] Pack forward-only FASTA... 0.00 sec</span></span>
<span><span class="co"># [bwa_index] Construct SA from BWT and Occ... 0.00 sec</span></span>
<span><span class="co"># [main] Version: 0.7.17-r1188</span></span>
<span><span class="co"># [main] CMD: bwa index consensus.fa</span></span>
<span><span class="co"># [main] Real time: 0.004 sec; CPU: 0.004 sec</span></span>
<span><span class="co"># [M::bam2fq_mainloop] discarded 228 singletons</span></span>
<span><span class="co"># [M::bam2fq_mainloop] processed 27228 reads</span></span>
<span><span class="co"># [M::bwa_idx_load_from_disk] read 0 ALT contigs</span></span>
<span><span class="co"># [M::process] read 27000 sequences (4050000 bp)...</span></span>
<span><span class="co"># [M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1190, 0, 0)</span></span>
<span><span class="co"># [M::mem_pestat] skip orientation FF as there are not enough pairs</span></span>
<span><span class="co"># [M::mem_pestat] analyzing insert size distribution for orientation FR...</span></span>
<span><span class="co"># [M::mem_pestat] (25, 50, 75) percentile: (348, 429, 530)</span></span>
<span><span class="co"># [M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 894)</span></span>
<span><span class="co"># [M::mem_pestat] mean and std.dev: (422.49, 140.29)</span></span>
<span><span class="co"># [M::mem_pestat] low and high boundaries for proper pairs: (1, 1076)</span></span>
<span><span class="co"># [M::mem_pestat] skip orientation RF as there are not enough pairs</span></span>
<span><span class="co"># [M::mem_pestat] skip orientation RR as there are not enough pairs</span></span>
<span><span class="co"># [M::mem_process_seqs] Processed 27000 reads in 1.101 CPU sec, 1.064 real sec</span></span>
<span><span class="co"># [main] Version: 0.7.17-r1188</span></span>
<span><span class="co"># [main] CMD: bwa mem consensus.fa tempR1.fastq tempR2.fastq</span></span>
<span><span class="co"># [main] Real time: 1.097 sec; CPU: 1.133 sec</span></span>
<span><span class="co"># result file temp.sam is here: temp.sam</span></span>
<span></span>
<span><span class="va">sam_file</span></span>
<span><span class="co"># [1] "temp.sam"</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/count_tryptase.html">count_tryptase</a></span><span class="op">(</span><span class="va">sam_file</span><span class="op">)</span></span>
<span><span class="va">counts</span></span>
<span><span class="co"># [1]  24   0  22   0  26   0 347 316 304</span></span>
<span></span>
<span><span class="fu"><a href="reference/TrypLik.html">TrypLik</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span>
<span><span class="co">#   Alpha_count Beta_count Beta_FS_count Posterior_likelihood</span></span>
<span><span class="co"># 1           2          2             0              0.99890</span></span>
<span><span class="co"># 2           2          3             0              0.00008</span></span>
<span><span class="co"># 3           3          2             0              0.00082</span></span>
<span><span class="co"># 4           3          3             0              0.00020</span></span></code></pre></div>
<p>(the <code>TrypLik</code> function takes either a single vector of counts or values provided to specific arguments).</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Below is the citation output from using <code>citation('TrypLik')</code> in R. Please run this yourself to check for any updates on how to cite <strong>TrypLik</strong>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"TrypLik"</span><span class="op">)</span>, bibtex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in citation("TrypLik"): no date field in DESCRIPTION file of package</span></span>
<span><span class="co">#&gt; 'TrypLik'</span></span>
<span><span class="co">#&gt; Warning in citation("TrypLik"): could not determine year for 'TrypLik' from</span></span>
<span><span class="co">#&gt; package DESCRIPTION file</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To cite package 'TrypLik' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Carroll J, Wall J (????). _TrypLik: Calculates Tryptase Variant</span></span>
<span><span class="co">#&gt;   Likelihoods_. R package version 0.1.0.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {TrypLik: Calculates Tryptase Variant Likelihoods},</span></span>
<span><span class="co">#&gt;     author = {Jonathan Carroll and Jeff Wall},</span></span>
<span><span class="co">#&gt;     note = {R package version 0.1.0},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
<p>Please note that the <code>TrypLik</code> was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the <code>TrypLik</code> project is released with a <a href="http://bioconductor.org/about/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div class="section level2">
<h2 id="development-tools">Development tools<a class="anchor" aria-label="anchor" href="#development-tools"></a>
</h2>
<ul>
<li>Package development is possible thanks to <em><a href="https://CRAN.R-project.org/package=usethis" class="external-link">usethis</a></em>, <em><a href="https://CRAN.R-project.org/package=remotes" class="external-link">remotes</a></em>, and <em><a href="https://CRAN.R-project.org/package=rcmdcheck" class="external-link">rcmdcheck</a></em> customized to use <a href="https://www.bioconductor.org/help/docker/" class="external-link">Bioconductor’s docker containers</a> and <em><a href="https://bioconductor.org/packages/3.16/BiocCheck" class="external-link">BiocCheck</a></em>.</li>
<li>The <a href="http://HIBio.github.io/TrypLik" class="external-link">documentation website</a> is automatically updated thanks to <em><a href="https://CRAN.R-project.org/package=pkgdown" class="external-link">pkgdown</a></em>.</li>
<li>The code is styled automatically thanks to <em><a href="https://CRAN.R-project.org/package=styler" class="external-link">styler</a></em>.</li>
<li>The documentation is formatted thanks to <em><a href="https://CRAN.R-project.org/package=devtools" class="external-link">devtools</a></em> and <em><a href="https://CRAN.R-project.org/package=roxygen2" class="external-link">roxygen2</a></em>.</li>
</ul>
<!-- For more details, check the `dev` directory. --><p>This package was developed using <em><a href="https://bioconductor.org/packages/3.16/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing TrypLik</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jonathan Carroll <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1404-5264" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jeff Wall <br><small class="roles"> Author </small>  </li>
<li>Human Immunology Biosciences (HI·Bio) <br><small class="roles"> Copyright holder, funder </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/HIBio/TrypLik/issues" class="external-link"><img src="https://img.shields.io/github/issues/HIBio/TrypLik" alt="GitHub issues"></a></li>
<li><a href="https://github.com/HIBio/TrypLik/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/HIBio/TrypLik" alt="GitHub pulls"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jonathan Carroll, Jeff Wall, Human Immunology Biosciences (HI·Bio).</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
