---
title: "Estimating Tryptase Copy Numbers from Sequence Data"
author: 
  - name: Jeff Wall
    affiliation:
    - TBA
    email: TBA
  - name: Jonathan Carroll
    affiliation:
    - Human Immunology Biosciences
    email: jonathan@hibio.com
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
date: "`r doc_date()`"
package: "`r pkg_ver('TrypLik')`"
vignette: >
  %\VignetteIndexEntry{TrypLik}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
  %\VignetteDepends{ggplot2}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    crop = NULL ## Related to https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016656.html
)

required <- c("ggplot2")
if (!all(unlist(lapply(required, function(pkg) requireNamespace(pkg, quietly = TRUE)))))
  knitr::opts_chunk$set(eval = FALSE)
```

```{r vignetteSetup, echo=FALSE, message=FALSE, warning = FALSE}
## Track time spent on making the vignette
startTime <- Sys.time()

## Bib setup
library("RefManageR")

lyons <- bibentry(
  bibtype = "Article", 
  author = c(person(given = "Jonathan", middle = "J", family = "Lyons et al.")),
  title = "Elevated basal serum tryptase identifies a multisystem disorder associated with increased TPSAB1 copy number.",
  journal = "Nat Genet.",
  year = "2016", 
  volume = "48(12)",
  pages = "1564-1569", 
  doi = "doi:10.1038/ng.3696"
)
maun <- bibentry(
  bibtype = "Article",
  author = c(person(given = "Henry", middle = "R", family = "Maun et al.")),
  title = "An Allosteric Anti-tryptase Antibody for the Treatment of Mast Cell-Mediated Severe Asthma",
  journal = "Cell",
  year = "2019",
  volume = "179(2)",
  pages = "417-431",
  doi = "doi:10.1016/j.cell.2019.09.009"
)
glover <- bibentry(
  bibtype = "Article",
  author = c(person(given = "Sarah", middle = "C", family = "Glover"),
             person(given = "Alexander", family = "Carlyle"),
             person(given = "Jonathan", middle = "J", family = "Lyons")),
  title = "Hereditary alpha-tryptasemia despite normal tryptase-encoding gene copy number owing to copy number loss in trans",
  journal = "Annals of Allergy, Asthma & Immunology",
  year = "2022",
  volume = "128(4)",
  pages = "460-461",
  doi = "10.1016/j.anai.2021.12.008"
)
chollet <- bibentry(
  bibtype = "Article",
  author = c(person(given = "Madeleine", middle = "B", family = "Chollet"),
             person(given = "Anna", family = "Kovalszki"),
             person(given = "Cem", family = "Akin")),
  title = "A novel case of beta-tryptase/TPSB2 allele duplication",
  journal = "Journal of Allergy and Clinical Immunology",
  year = "2021",
  volume = "147(2) supplement",
  doi = "10.1016/j.jaci.2020.12.491"
)
chovanec <- bibentry(
  bibtype = "Unpublished",
  author = "Chovanec",
  title = "Unpublished data",
  note = "TBA",
  year = "2022"
)

## Write bibliography information
bib <- c(
    R = citation(),
    BiocStyle = citation("BiocStyle")[1],
    biocthis = citation("biocthis")[1],
    knitr = citation("knitr")[1],
    RefManageR = citation("RefManageR")[1],
    rmarkdown = citation("rmarkdown")[1],
    sessioninfo = citation("sessioninfo")[1],
    testthat = citation("testthat")[1],
    TrypLik = citation("TrypLik")[1],
    lyons,
    maun,
    glover,
    chollet,
    chovanec
)

BibOptions(max.names = 1, longnamesfirst = FALSE, cite.style = "authoryear", bibpunct = c("", "", "", "", ";", ","))
```

<!-- # Basics -->

<!-- ## Install `TrypLik` R package -->

<!-- The R package can be installed with the following: -->

<!-- ```{r "install", eval = FALSE} -->
<!-- if (!requireNamespace("BiocManager", quietly = TRUE)) { -->
<!--       install.packages("BiocManager") -->
<!--   } -->

<!-- BiocManager::install("TrypLik") -->

<!-- ## Check that you have a valid Bioconductor installation -->
<!-- BiocManager::valid() -->
<!-- ``` -->

<!-- # Command-line usage -->

<!-- Additionally, the C code which performs the internal aggregation is provided in  -->
<!-- `inst/tryplik.c` for use within a pipeline outside of R. -->


```{css, echo=FALSE}
.bgr {
  background-color: salmon;
  padding: 0;
}
.bgy {
  background-color: yellow;
  padding: 0;
}
.bgg {
  background-color: springgreen;
  padding: 0;
}
.bgp {
  background-color: pink;
  padding: 0;
}
.tg {
  background-color: unset;
  font-weight: bold;
  padding: unset;
  color: green
}
.bgx {
  background-color: silver;
  padding: 0;
}
.mono {
  font-family: monospace;
  font-size: 120%;
}
.lefttable table {
  margin-left: 50px;
  max-width: 700px;
}
.superwide table {
  margin-left: 50px;
  max-width: 1000px;
  width: 120%;
}
.evensuperwider table {
  margin-left: 50px;
  max-width: 1600px;
  width: 150%;
}
```

# Overview 

We have developed a likelihood-based approach for estimating tryptase copy
numbers (specifically the numbers of $\alpha$ alleles, $\beta$ alleles and $\beta$
frameshift alleles) directly from whole-genome (preferable) or whole-exome
sequence data. The approach utilizes an updated assembly of canonical $\alpha$,
$\beta$, and $\delta$ tryptase alleles, and identifies reads that map uniquely
to only one of these alleles. The complete workflow is implemented in the
package TrypLik.

# Background

The tryptase gene cluster contains four closely spaced paralogous genes on
chromosome 16p13.3. These include TPSAB1 and TPSB2, which together encode the
$\alpha$ and $\beta$ tryptase peptides. TPSAB1 encodes two major alleles,
$\alpha$ and $\beta$I, while TPSB2’s major alleles are $\beta$II and $\beta$III.
$\alpha$ and $\beta$I are $\sim1\%$ diverged over a $1.2$ Kbp region, while
$\beta$I and $\beta$II differ by just a single nucleotide over the same region.
$\alpha$ alleles are thought to be enzymatically inactive, and while most
$\beta$ alleles are active, there is a specific $\beta$ frameshift variant
($\beta$III$^{FS}$) that is inactive.

Recent work has identified a familial inherited syndrome, hereditary alpha
tryptasemia (H$\alpha$T), which is associated with one or more duplications of
the $\alpha$ allele in TPSAB1 `r AutoCite(lyons)`. Common phenotypes found in
H$\alpha$T patients include systemic venom reactions, pruritis, IBS, reflux,
skeletal abnormalities, retained primary dentition and anthralgia. There is
also evidence for $\beta$ duplications and deletions (`r AutoCite(chollet)`,
`r AutoCite(glover)`), though it is not yet clear if this reflects copy number
variants (CNVs) in only TPSAB1 or CNVs in both TPSAB1 and TPSAB2. Nothing is
known about the clinical effects of tryptase $\beta$ copy number variation.

While $\alpha$, $\beta$, and $\beta$III$^{FS}$ copy numbers can be accurately
determined using custom droplet digital PCR (ddPCR) assays (cf. `r AutoCite(lyons)`), 
it is much more challenging to estimate copy numbers from
standard short-read (e.g. Illumina) sequence data, even with a high depth of
coverage. This is primarily due to the fact that $\alpha$, $\beta$, and
$\beta$III$^{FS}$ tryptase alleles are so similar, but exacerbated by errors in
the human reference genome assembly (hg38), which appears to have chimeric
alleles for both TPSAB1 and TPSB2. In addition, the published allelic sequences
in `r AutoCite(maun)` also appear to be incorrect. This has motivated our
efforts to generate our own consensus assemblies of different tryptase alleles
(see below).

# Instructions on use

There are two main parts to the workflow: First, we map reads to a tryptase
consensus sequence and count the ones that map uniquely to tryptase $\alpha$,
$\beta$, $\beta$III$^{FS}$ and $\delta$ alleles. Second, we perform Bayesian
likelihood calculations to estimate the most likely copy number count. We
describe each step in greater detail below.

## Tryptase read mapping

This part of the workflow can be broken up into the following components:

1. Subset an indexed bam file to consider only reads mapping to the tryptase region
1. Convert the output back to (the original) paired-end fastq reads
1. Map raw reads to consensus tryptase sequence
1. Remove duplicates
1. Count the numbers of reads containing specific 30 bp motifs that map specifically to a single tryptase allele 

There are different ways to go about each of these steps:

### Indexing bam files

If a datafile is not already indexed, we recommend using samtools to do so. For
example, we use publicly available high-coverage whole genome sequence data from
the 1000 Genomes Project as a test case. We downloaded the file
"HG00100.final.cram" from
https://www.internationalgenome.org/data-portal/sample/HG00100 to use as an
example.Indexing and subsetting can be performed with the following commands:

```bash
samtools index HG00100.final.cram
samtools view -b -h HG00100.final.cram chr16:1200000-1300000 > HG00100.tryptase.cram
samtools index HG00100.tryptase.cram
```

### Converting indexed cram files to fastq reads

One possibility for doing this uses samtools again:

```bash
samtools sort -n HG00100.tryptase.cram -o HG00100.tryptase.sorted.cram
samtools fastq -1 HG00100.tryptase.R1.fastq -2 HG00100.tryptase.R2.fastq -0 /dev/null -s /dev/null -n HG00100.tryptase.sorted.cram
```

### Mapping reads to tryptase consensus sequence 

Here, we are using the tryptase consensus sequence published by `r AutoCite(maun)`
to winnow out reads that are in the tryptase region that do not actually
map to tryptase genes. The following command works for our continuing example:

```bash
bwa mem consensus.fa HG00100.tryptase.R1.fastq HG00100.tryptase.R2.fastq > out.sam
```

### Removing duplicates

This is generally done using Picard or samtools. Here is an example of doing so
using samtools, with the output written to the file "temp.sam".

```bash
samtools sort -n out.sam -o out.sorted.sam
samtools fixmate -m out.sorted.sam fixmate.sam
samtools sort -o positionsort.sam fixmate.sam
samtools markdup -r positionsort.sam temp.sam
```

### Counting reads mapping to specific tryptase alleles

The file "count_tryptase.sh" is a shell script that counts the numbers of reads
that map specifically to different tryptase alleles (see below for technical
details). It requires the input file to be called "temp.sam". Output consists
of $9$ integers summarizing read counts:

* The first four of these are the numbers of reads mapping uniquely to $\alpha$,
$\beta$I, $\beta$II or $\beta$III alleles in the specific $83$ bp region where
all four alleles can be distinguished from each other.

* The next two are the numbers of reads that map uniquely to $\beta$ ‘wildtype’
(i.e. not frameshifted) and $\beta$ frameshifted sequences

* The final three are the numbers of reads that can be uniquely mapped to
$\alpha$, $\beta$, or $\delta$ alleles respectively

Further details on these can be found in the technical details below. For our
example with sample HG00100 from the 1000 Genomes Project, the output of
"count_tryptase" is

```bash
24    0   22    0   26    0   347   316   304
```

## Tryptase copy number likelihood calculations 

The last five numbers from the output of "count_tryptase" are used as input to
the program TrypLik to estimate the likelihoods of different combinations of
total number of $\alpha$ alleles, total number of $\beta$ alleles, and total
number of $\beta$ frameshift alleles. The method accounts for dependence across
read counts (caused by the same $2 \times 150$ bp read generally overlapping
multiple of the regions used to distinguish between alleles) and also uses prior
information on CNV frequencies in different populations. Usage is

```bash
TrypLik WT FS Areads Breads Dreads (-pop POP)
```

`WT`: number of reads mapping to $\beta$ "wildtype" alleles at the site of the
frameshift mutation (5th field from "count_tryptase" output)

`FS`: number of reads mapping to $\beta$ frameshift alleles at the site of the
frameshift mutation (6th field from "count_tryptase" output)

`Areads`: number of reads mapping uniquely to $\alpha$ alleles relative to
$\beta$ and $\delta$ alleles (7th field from "count_tryptase" output)

`Breads`: number of reads mapping uniquely to $\beta$ alleles relative to
$\alpha$ and $\delta$ alleles (8th field from "count_tryptase" output)

`Dreads`: number of reads mapping uniquely to $\delta$ alleles relative to
$\alpha$ and $\beta$ alleles (9th field from "count_tryptase" output)

`-pop POP`: optional flag to specify the predominant ancestry of the test
sample. Options include EUR for European ancestry, SAS for South Asian
ancestry, EAS for East Asian ancestry and AFR for African ancestry. The default,
if the `-pop` flag is not included, is European ancestry.

Note that these likelihood calculations assume that all individuals contain
exactly two tryptase $\delta$ alleles. `Dreads` acts as a ‘control’ from which
we estimate the other copy numbers. The output consists of a list of different
$\alpha$, $\beta$ and $\beta$ frameshift allele copy numbers followed by the
combination’s posterior probability. To continue our example from before,
calculation for HG00100 would use the command

```bash
TrypLik 26 0 347 316 304 -pop EUR
```

Output would look like this:

| Alpha count | Beta count | Beta FS count | Posterior likelihood |
|:------------|:-----------|:--------------|:---------------------|
|      2      |      2     |       0       |       0.99890        |
|      2      |      3     |       0       |       0.00008        |
|      3      |      2     |       0       |       0.00082        |
|      3      |      3     |       0       |       0.00020        |

The method estimates that HG00100 has a 2$\alpha:2\beta:2\delta$ genotype
(with no frameshift alleles) with posterior probability 99.89%.

# Technical details

## de novo assembly/alignment

We started by aligning cDNA sequences to a 40 bp region near the probes used by
`r AutoCite(lyons)`(Supplementary Figure 3b) for ddPCR.

<div class="lefttable">
|   |      |
|---|:-----|
| Delta (CCDS10432.1) | <span class="mono">CCAGGCCAGG CCCTGCAGCA A<mark class="bgr">A</mark>CGGGCAT<mark class="bgr">T</mark> GTTGGGGG<mark class="bgr">G</mark>C</span> |
| Alpha (M30038.1)  | <span class="mono">CCAG<mark class="bgy">T</mark>CCAGG CCCTGCAGC<mark class="bgy">A</mark> AG<mark class="bgy">C</mark>GGG<mark class="bgy">T</mark>ATC GT<mark class="bgy">C</mark>GGGGGTC</span> |
| Beta1 (M33492.1)   | <span class="mono">CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark> AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC GT<mark class="bgy">C</mark>GGGGGTC</span> | 
| Beta2 (BC065923.1)  | <span class="mono">CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark> AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC GT<mark class="bgy">T</mark>GGGGGTC</span> |
| Beta3 (AF099143)  | <span class="mono">CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark> AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC GT<mark class="bgy">T</mark>GGGGGTC</span> |
</div>

Here sites in red are fixed differences between TPSD1 and (TPSAB1+TPSB2), while
sites in yellow can be used to distinguish $\alpha$, $\beta$I, and
$\beta$II/$\beta$III alleles from each other. $\beta$II and $\beta$III alleles
are distinguishable from each other at sites near the `r AutoCite(lyons)` REV
primer as follows:

|   |      |
|---|:-----|
| Beta2 (BC065923.1) | <span class="mono">GAGAGTCC<mark class="bgy">A</mark>C G<mark class="bgy">G</mark>CC<mark class="bgy">C</mark>ATACT</span> |
| Beta3 (AF099143)   | <span class="mono">GAGAGTCC<mark class="bgy">G</mark>C G<mark class="bgy">A</mark>CC<mark class="bgy">G</mark>ATACT</span> |

We assumed that the sequences listed above are correct and invariant in the
different tryptase alleles. We then used Illumina short read data to ‘extend’
allele-specific alignments. Specifically, we combined the raw reads from 30
randomly chosen individuals of European ancestry from the 1000 Genomes Project
(high-coverage data generated by the NYGC) and pulled out those reads that
matched the above sequences. We then could expand on the allele-specific
assemblies by looking at the alignments just 5’ and 3’ to the sequences shown
above, and concurrently identify common allele-specific polymorphisms. We
continued to iteratively extend these assemblies until $\alpha$ and $\beta$
alleles could not be distinguished from each other. Roughly speaking, since the
Illumina data consisted of 150 bp reads, this process can continue as long as
there is a fixed difference between allelic assemblies every $\sim 150$ bp. We
extended these assemblies across $\sim 1.1$ Kbp of sequence including most of the
coding regions (see Appendix 1).

Our alignments show the expected difference between $\alpha$ and $\beta$ alleles
at the putative functional mutation. In addition, we confirm that the
frameshift mutation happens only on a $\beta$III background (leading to the
inactivated $\beta$III$^{FS}$ allele), and that the $\beta$I and $\beta$II consensus
sequences differ by just a single nucleotide, despite $\beta$I being an allele
that is (supposed to be) exclusive to TPSAB1 and $\beta$II being confined to
TPSB2. This likely reflects paralogous gene conversion between tryptase genes
over evolutionarily recent time scales.

The tryptase allele assemblies generated here differ slightly from all
previously published assemblies, including some of the cDNA sequences, the `r AutoCite(lyons)` 
tryptase alignment, the `r AutoCite(maun)` tryptase alignment, and
the human reference genome (hg38). We believe though that Illumina reads are
highly accurate, and can show how these previous assemblies are inconsistent
with Illumina data from multiple individuals.

# Tryptase CNV genotyping from high-coverage Illumina reads

From the alignment described above, we identified $11$ different $30$ bp regions
where reads could be uniquely mapped to $\alpha$, $\beta$, or $\delta$ alleles.
Using the base numbering defined by `r AutoCite(maun)`(Data S1) and also used in
the accompanying table, these correspond to the following regions:

```{=latex}
\begin{array}{ccc}
560\, – 589 & 1010\, – 1039 & 1360\, – 1389 \\
740\, – 769 & 1115 \sim 1146 & 1510\, – 1539 \\
840\, – 869 & 1220\, – 1249 & 1560\, – 1589 \\
940\, – 969 & 1260\, – 1289 
\end{array}
```

Tallying up the total number of ‘$\alpha$ reads’, ‘$\beta$ reads’ and ‘$\delta$
reads’ (which we call here $A$, $B$ and $D$ respectively) gives us a way of estimating
the relative copy numbers of $\alpha$, $\beta$ and $\delta$ alleles. In
particular, if we assume each individual has exactly two copies of TPSD1 (i.e.
no copy number variation in TPSD1), then the number of $\alpha$ and $\beta$
alleles can be estimated by $2A/D$ and $2B/D$ respectively.

Some caveats apply:

1. The regions are not independent of each other, since the $2 \times 150$ bp paired-end
reads generally include multiple regions. This leads to a greater variance than
otherwise expected in estimated allele counts.

1. Even in a pool of individuals with the same true number of $\alpha$ and $\beta$
alleles, the estimates $2A/D$ and $2B/D$ will be correlated since they both have the
same denominator.

1. Unknown (or unaccounted for) polymorphisms within an allelic class can lead to
misclassified reads. This is perhaps most important in the analysis of
non-European ancestry individuals.

1. The method assumes that all CNVs affect the entire $1$ Kbp test region, though we
know this isn’t true. `r AutoCite(chovanec)` report the existence of σ alleles,
which appear to be a chimera of $\alpha$ and $\beta$ alleles. In our analyses,
these alleles are estimated as partial $\alpha$ and partial $\beta$,
proportional to the numbers of reads mapping to each canonical allele assembly.

We suggest using an empirically-based approach for estimating the bias in the
estimates (e.g. estimating 

$$
E \bigl( \frac{2A}{D} |\ \textrm{actual number of }\alpha\textrm{ alleles is known} \bigr)
$$

that uses orthogonal information on true allele counts (e.g. validated
ddPCR data). A preliminary version of this is described below.

# $\beta$III frameshift mutation

The approach described above can estimate the total copy number of tryptase beta
alleles, but some of these beta alleles are inactive due to a frameshift
mutation. In parallel, we conduct a read-counting exercise that counts reads
that uniquely map to the $\beta$III$^{FS}$ allele and reads that map to all other
$\beta$ alleles:

|   |      |
|---|:-----|
| $\beta$III$^{FS}$ | <span class="mono">CTCAGAGACC TTCCCCCCCG GGGATGCCGT</span> |
| All other $\beta$ | <span class="mono">CTCAGAGACC TTCCCCCC&ndash;G GGGATGCCGTG</span> |
 
(Note that both $\alpha$ and $\sigma$ alleles do not map to either of these sequences.)
The observed numbers of reads in each of these two categories (called here FS+
and FS-) is expected to follow a binomial distribution with probabilities
proportional to the true numbers of $\beta$III$^{FS}$ and non-$\beta$III$^{FS}$
$\beta$ alleles. Further, by assumption, $\beta$III$^{FS}$ alleles only occur
in TPSB2. If data = \{FS+, FS-\}, then under a simple likelihood framework,

$$
\rm{lik} \bigl(data\bigr) \sim Pr \bigl(genotype\bigr)\ Pr \bigl( data\ |\ genotype\bigr)
$$

Suppose for example we observe $28$ and $36$ reads supporting $\beta$III$^{FS}$
and
non-$\beta$III$^{FS}$ $\beta$ alleles respectively (i.e. data = $\{28, 36\}$) 
in an individual of European ancestry with an estimated tryptase CNV
genotype of $0\alpha:4\beta:2\delta$. Based on the European $1$ KGP data, the
frequency of the $\beta$III$^{FS}$ mutation is $\sim\!0.1766$. So, the naïve
(i.e. prior or Hardy-Weinberg) probability of a homozygous $\beta$III$^{FS}$
genotype (i.e. $\beta$III$^{FS}$ mutations on both chromosomes) is $(0.1766)^2$
and the probability of having exactly one copy of $\beta$III$^{FS}$ is
$2(0.1766)(1-0.1766)$. Assuming the estimated tryptase CNV genotype is correct,
we calculate

$$
 \textrm{Pr} \bigl(\textrm{data}\ |\ \beta\textrm{III}^{FS}\, \textrm{heterozygote}\bigr) = {64 \choose 28} (0.25)^{28}(0.75)^{36},\ \textrm{and}
$$
$$
 \textrm{Pr} \bigl(\textrm{data}\ |\ \beta\textrm{III}^{FS}\, \textrm{homozygote}\bigr) = {64 \choose 28} (0.5)^{28} (0.5)^{36}
$$

and 

$$
 \textrm{lik} \bigl(\beta\textrm{III}^{FS}\, \textrm{homozygote}\bigr) \sim 0.0018922,\ \textrm{and}
$$
$$
 \textrm{lik} \bigl(\beta\textrm{III}^{FS}\, \textrm{heterozygote}\bigr) \sim 0.00014354
$$

In a Bayesian context, the posterior 
$\textrm{Pr} \bigl(\beta\textrm{III}^{FS}\, \textrm{homozygote}\bigr) = 0.9295$, 
and we conclude that the individual is
likely to have $0 \alpha$ alleles, $2$ active $\beta$ alleles and $2$ inactive
$\beta$III$^{FS}$ alleles.

# Tryptase CNV genotyping revisited: likelihood framework

In an idealized world where the 4 caveats described above do not apply, each
read would be independent and the read counts $(A, B, D)$ could be modeled as
following a multinomial distribution conditional on the sum $A+B+D$, with
probabilities for each category proportional to the true copy numbers of
$\alpha$ $\beta$ and $\delta$. We employ two modifications to this model to
account for deviations from model assumptions:

1. We replace $(A+B+D)$ by $(A+B+D) / R$, the "effective number of independent
read counts", to account for caveat #1

1. We reweight the multinomial probabilities by factors of 
$E(2A/D\ |\ \alpha\ \textrm{copy number} = 1)$ and $E(2B/D\ |\ \beta\ \textrm{copy number} = 3)\ /\ 3$ 
for $\alpha$ and $\beta$ copy numbers to account for unknown structural variants and/or
polymorphisms

We use empirical data from the EUR populations from the 1000 Genomes Project to
estimate these correction factors. First, we plotted $2A/D$ vs. $2B/D$ for these $503$
samples:

```{r, genomes, class.source="fold-hide"}
data("genomes", package = "TrypLik")

library(ggplot2)

ggplot(genomes, aes(`2A/D`, `2B/D`, col = `Geno Group`)) + 
  geom_point() +
  theme_bw() +
  theme(text = element_text(size = 16))
```

We focus on the dots colored green, which are estimated to have a tryptase
genotype of $1\alpha:3\beta:2\delta$. The average values of $2A/D$ and
$2B/D$ are $1.06$ and $3.13$ respectively, and these means are relatively
insensitive to the inclusion/exclusion of specific points on the periphery of
the central mass. In addition, we note that

$$
\begin{eqnarray}
 \textrm{Var}(2A/D) &=& 0.0263 \\
 \textrm{Var}(2B/D) &=& 0.1286 \\
 r^2 (2A/D, 2B/D) &=& 0.1953 \\
\end{eqnarray}
$$

We then ran simulations with different values of R (but conditional on the
observed $A$, $B$ and $D$ values) to heuristically find a value that, on
average, produces correlation and variances similar to what was observed. We
settled on $R = 2.75$, but with the caveat that this estimation process was
entirely qualitative. We do note though that this $R$ is approximately equal to
the number of $30$ bp informative regions covered by an average $2\times 150$ bp
read pair in the target tryptase region. As an example of the process, suppose
the observed data $\{A, B, D\} = \{192, 600, 290\}$ and that we want to estimate
the probability of observing the data given a true genotype of
$1\alpha:4\beta$. Then, $2A/D = 1.324, 2B/D = 4.138$, and the ‘effective
number of independent reads is $(192+600+290)/2.75 \approx 393$. Then, to estimate the
probability of the genotype $1\alpha:3\beta:2\delta$, for example, we
calculate

$$
\begin{align}
&\textrm{Pr}\left(2A/D \approx 1.324\ \&\&\ 2B/D \approx 4.138\ |\ \textrm{true genotype is } 1\alpha:3\beta:2\delta\right) \approx \\
&\textrm{Pr}\left(2A/D \approx 1.324\ \&\&\ 2B/D \approx 4.138\ |\ 393 \textrm{ independent reads are simulated from a}\right. \\
&\textrm{multinomial distribution with relative probabilities of } 1.06, 3.13 \textrm{ and } 2 \textrm{ for } \alpha, \beta, \\
&\left.\textrm{ and } \delta \textrm{ reads respectively}\right)
\end{align}
$$
These are actually approximate likelihoods; operationally, $2A/D \approx 1.324$
is the estimated probability that the simulated $2A/D$ is between
$(0.99\times1.324)$ and $(1.01\times1.324)$. To calculate posterior
probabilities, we still need a prior estimate of the population-level genotype
frequencies. There is some element of circularity here since we ultimately are
estimating genotypes (which go in to the population-level genotype frequency
estimates). Ideally we would have an independent way of assessing population
frequencies (e.g. widespread ddPCR), but in the absence of that we have
obtained crude prior estimates (see Table 1 below) by qualitative assessments of
figures like Figure 1. Continuing with our example (and assuming European
ancestry), we estimate

$$
\begin{align}
  \textrm{lik}\left(1\alpha:3\beta:2\delta\right) &\sim \textrm{Pr}\left(1\alpha:3\beta:2\delta\right)\ \times \\
  &\; \; \; \; \; \textrm{Pr}\left(2A/D \approx 1.324\ \&\&\ 2B/D \approx 4.138\ | \right. \\ 
  &\; \; \; \; \; \; \; \; \; \; \; \left.\textrm{genotype} = 1\alpha:3\beta:2\delta\right) \\
  &= 0.434 \times 0.0002521 \\
  \\
  \textrm{lik}\left(1\alpha:4\beta:2\delta\right) &\sim \textrm{Pr}\left(1\alpha:4\beta:2\delta\right)\ \times \\ 
  &\; \; \; \; \; \textrm{Pr}\left(2A/D \approx 1.324\ \&\&\ 2B/D \approx 4.138\ | \right. \\ 
  &\; \; \; \; \; \; \; \; \; \; \; \left. \textrm{genotype} = 1\alpha:4\beta:2\delta\right) \\
  &= 0.03 \times 0.0012967 \\
  \\
  \textrm{lik}\left(1\alpha:5\beta:2\delta\right) &\sim \textrm{Pr}\left(1\alpha:5\beta:2\delta\right)\ \times \\
  &\; \; \; \; \; \textrm{Pr}\left(2A/D \approx 1.324\ \&\&\ 2B/D \approx 4.138\ | \right. \\ 
  &\; \; \; \; \; \; \; \; \; \; \; \left. \textrm{genotype} = 1\alpha:5\beta:2\delta\right) \\
  &= 0.0004 \times 0.000019 
\end{align}
$$

The posterior probability of a $1\alpha:3\beta:2\delta$ genotype is estimated at 

$$
\frac{0.434 \times 0.0002521}{0.434 \times 0.0002521 + 0.03 \times 0.0012967 + 0.0004 \times 0.000019} = 0.738
$$

Note that this estimate (and many others) is highly sensitive to the (unknown)
prior population frequencies. $\textrm{Pr}\left(\textrm{data}\ |\ \textrm{genotype}\right)$ 
is highest for $1\alpha:4\beta:2\delta$ by a factor of
$5$, but this is swamped out by the very low prior for the allele frequency.

Table 1. Prior estimates of tryptase genotype frequencies, stratified by
ancestry: EUR = European, EAS = East Asian, AFR = African, SAS = South Asian

<div class="lefttable">
| $\alpha$ copy number | $\beta$ copy number | EUR | EAS | AFR | SAS |
|:----------------------|:--------------------|:----|:----|:----|:----|
|0|2|0.0001|0.025|0.001|0.004|
|0|3|0.013|0.037|0.015|0.053|
|0|4|0.219|0.032|0.274|0.165|
|0|5|0.020028|0.004|0.1|0.0325|
|0|6|0.0004|0.0002|0.008|0.0006|
|0|7|0.00005|0.00002|0.0003|0.00005|
|0|8|0.00001|0.000002|0.0001|0.000005|
|1|1|0.0001|0.0021|0.0001|0.0005|
|1|2|0.013|0.22|0.012|0.017|
|1|3|0.434|0.23|0.39|0.444|
|1|4|0.03|0.012|0.06|0.01|
|1|5|0.0004|0.0005|0.005|0.0005|
|1|6|0.00001|0.00001|0.0001|0.00005|
|1|7|0.000001|0.000003|0.00001|0.000005|
|2|1|0.001|0.004|0.0007|0.0003|
|2|2|0.208|0.42|0.126536|0.256129|
|2|3|0.03|0.005|0.003|0.004|
|2|4|0.004|0.0005|0.0001|0.001|
|2|5|0.0001|0.00001|0.00001|0.0001|
|2|6|0.00004|0.000002|0.000001|0.00001|
|3|1|0.0001|0.0001|0.0001|0.0001|
|3|2|0.018|0.006|0.003|0.01|
|3|3|0.005|0.001|0.0003|0.0003|
|3|4|0.0004|0.0001|0.0001|0.0001|
|3|5|0.00001|0.00001|0.0001|0.00001|
|4|2|0.002|0.0001|0.0001|0.0003|
|4|3|0.0008|0.0001|0.0001|0.0001|
|4|4|0.00005|0.0001|0.0001|0.0001|
|4|5|0.00001|0.0001|0.0001|0.0001|
|5|2|0.0002|0.00001|0.00001|0.00001|
|5|3|0.0001|0.00001|0.00001|0.00001|
|5|4|0.00004|0.00001|0.00001|0.00001|
|5|5|0.00001|0.00001|0.00001|0.00001|
|6|2|0.00003|0.000001|0.000001|0.00001|
|6|3|0.00001|0.000001|0.000001|0.000001|
|6|4|0.000001|0.000001|0.000001|0.000001|
</div>

# Appendix

Tryptase region consensus alignments for different alleles, using the numbering of `r AutoCite(maun)`

<mark class="tg">Green text</mark> = cDNA sequence is potentially incorrect

<mark class="bgr">Red highlight</mark> = fixed difference between delta and (alpha + beta)

<mark class="bgy">Yellow highlight</mark> = polymorphic within alpha + beta, and
can be used to discriminate between alpha &/or beta alleles

<mark class="bgx">Grey highlight</mark> = polymorphic within the specific allele/gene

### Bases 370 -- 449

<div class="superwide">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">GCCCGTCCTG GCGAGCC<mark class="bgx">CG</mark>G CCTACG<mark class="bgx">T</mark>GGC CCCTGGTGAG TCCCAGCC<mark class="bgx">G</mark>G GGTCCACCCT GCCCCTCACC ACATTCCACA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">GCCCGTCCTG GCGAGCC<mark class="bgx">GC</mark>G CCTACGCGGC CCCTGGTGAG TCCCAGCCGG GGTCCACCCT GCCCCTCACC ACATTCCACA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">GCCCGTCCTG GCGAGCCGCG CCTACG<mark class="bgx">C</mark>GGC CCCTGGTGAG TCCCAGCC<mark class="bgx">G</mark>G GGTCCACCCT GCCCCTCACC ACATTCCACA</span></td>
    </tr>
    </table>
</div>

### Bases 450 -- 549 

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">GATCAGGGCC TGGGTGGGTT CTGGGGAGGC CGGGCTGGCC CCCACACAGG GAAGGGCTGG <mark class="bgr">T</mark>CCCAGGCGT GGGGC<mark class="bgr">G</mark>GCTT C<mark class="bgr">T</mark>TGGTCCTG ACCTGGCACC</span></td>
      </tr>
      <tr style="border-top: 1px solid #ccc;">
        <td>alpha</td><td><span class="mono">GGTCAGGGCC TGGGTGGGTT CTGGGGAGGC <mark class="bgy">T</mark>GGGCTGGCC CCCACACAGG GAAGGG<mark class="bgx">G</mark>TGG GCCCAGGC<mark class="bgy">G</mark>T GGGGCTGCTT CCTGGTCCTG ACCTGGCACC</span></td>
      </tr>
      <tr style="border-top: 1px solid #ccc;">
        <td>beta1</td><td><span class="mono">G<mark class="bgx">G</mark>TCAGGGCC TGGGTGGGTT CTGGGGAGG<mark class="bgx">C</mark> <mark class="bgy">C</mark>GGGCTGGCC CCCACACAGG GAAGGGCTGG GCCCAGGC<mark class="bgx">G</mark>T GGGGCTGCTT CCTGGTCCTG ACCTGGCACC</span></td>
      </tr>
      <tr style="border-top: 1px solid #ccc;">
        <td>beta2</td><td><span class="mono">G<mark class="bgx">G</mark>TCAGGGCC TGGGTGGGTT CTGGGGAGG<mark class="bgx">C</mark> <mark class="bgy">C</mark>GGGCTGGCC CCCACACAGG GAAGGGCTGG GCCCAGGC<mark class="bgy">C</mark>T GGGGCTGCTT CCTGGTCCTG ACCTGGCACC</span></td>
      </tr>
      <tr style="border-top: 1px solid #ccc;">
        <td>beta3</td><td><span class="mono">G<mark class="bgx">G</mark>TCAGGGCC TGGGTGGGTT CTGGGGAGG<mark class="bgx">C</mark> <mark class="bgy">C</mark>GGGCTGGCC CCCACACAGG GAAGGGCTGG GCCCAGGC<mark class="bgy">C</mark>T GGGGCTGCTT CCTGGTCCTG ACCTGGCACC</span></td>
    </tr>
  </table>
</div>

### Bases 550 -- 649 

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono"><mark class="tg">T</mark>GCCCC<mark class="tg">A</mark>GCC CCAGGCCAGG CCCTGCAGCA A<mark class="bgr">A</mark>CGGGCAT<mark class="bgr">T</mark> GTTGGGGG<mark class="bgr">G</mark>C AG<mark class="bgx">G</mark>AGGCCCC CAGGAGCAAG TGGCCCTGGC AGGTGAGCCT GAGAGTCCGC</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono"><mark class="tg">T</mark>GCCCC<mark class="tg">A</mark>GCC CCAG<mark class="bgy">T</mark>CCAGG CCCTGCAGC<mark class="bgy">A</mark> AG<mark class="bgy">C</mark>GGG<mark class="bgy">T</mark>ATC GT<mark class="bgy">C</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC AGGTGAGCCT GAGAGTCC<mark class="bgy">G</mark>C</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta1</td><td><span class="mono">TGCCCCAGCC CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark> AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC GT<mark class="bgy">C</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC AGGTGAGCCT GAGAGTCC<mark class="bgy">A</mark>C</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta2</td><td><span class="mono"><mark class="tg">T</mark>GCCCC<mark class="tg">A</mark>GCC CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark> AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC GT<mark class="bgy">T</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC AGGTGAGCCT GAGAGTCC<mark class="bgy">A</mark>C</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta3</td><td><span class="mono">TGCCCCAGCC CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark> AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC GT<mark class="bgy">T</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC AGGTGAGCCT GAGAGTCC<mark class="bgy">G</mark>C</span></td>
    </tr>
  </table>
</div>

### Bases 650 -- 749 

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">GGCCCATACT GGATGCACTT CTGCGGGGGC TCC<mark class="bgx">C</mark>TCATCC ACCCCCAGTG GGTGCT<mark class="bgr">A</mark>ACC GC<mark class="bgx">G</mark>GCGCACT GC<mark class="bgx">G</mark>TGG<mark class="bgr">A</mark>ACC GTGAGTCTCC <mark class="bgr">T</mark>GGGGCCTGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">G<mark class="bgy">A</mark>CC<mark class="bgy">G</mark>ATACT GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC GC<mark class="bgy">G</mark>GCGCACT GC<mark class="bgy">C</mark>TGGGACC GTGAGTCTCC C<mark class="bgx">A</mark>GGGCCTGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta1</td><td><span class="mono">G<mark class="bgy">G</mark>CC<mark class="bgy">C</mark>ATACT GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC GC<mark class="bgy">A</mark>GCGCACT GC<mark class="bgy">G</mark>TGGGACC GTGAGTCTCC CGGGGCCTGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta2</td><td><span class="mono">G<mark class="bgy">G</mark>CC<mark class="bgy">C</mark>ATACT GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC GC<mark class="bgy">A</mark>GCGCACT GC<mark class="bgy">G</mark>TGGGACC GTGAGTCTCC CGGGGCCTGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta3</td><td><span class="mono">G<mark class="bgy">A</mark>CC<mark class="bgy">G</mark>ATACT GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC GC<mark class="bgx">A</mark>GCGCACT GC<mark class="bgx">G</mark>TGGGACC GTGAGTCTCC CGGGGCCTGG</span></td>
    </tr>
  </table>
</div>

### Bases 750 -- 849

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">AGGGGTGGGC AAGGGCTGGA TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG GAC<mark class="bgr">A</mark>TCAAGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">AGGGGTGGG<mark class="bgy">C</mark> AAGGGCTGGA TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG GACGTCAAGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta1</td><td><span class="mono">AGGGGTGGG<mark class="bgy">G</mark> AAGGGCTGGA TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG GACGTCAAGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta2</td><td><span class="mono">AGGGGTGGG<mark class="bgy">G</mark> AAGGGCTGGA TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG GACGTCAAGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta3</td><td><span class="mono">A<mark class="bgx">G</mark>GGGTGGG<mark class="bgy">G</mark> AAGGGCTGGA TGTGAGCCCT GGCTCCCGG<mark class="bgx">G</mark> TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG GACGTCAAGG</span></td>
    </tr>
  </table>
</div>

### Bases 850 -- 949 

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">ATCTGGCCGC CCTCAGGGTG CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC GTGCACCCAC AGTTCTACAT</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">ATCTGGCC<mark class="bgy">A</mark>C CCTCAGGGTG CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCC<mark class="bgx">G</mark>GTCAG CAGGATCATC GTGCACCCAC AGTTCTACA<mark class="bgy">T</mark></span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta1</td><td><span class="mono">ATCTGGCC<mark class="bgy">G</mark>C CCTCAGGGTG CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC GTGCACCCAC AGTTCTACA<mark class="bgy">C</mark></span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta2</td><td><span class="mono">ATCTGGCC<mark class="bgy">G</mark>C CCTCAGGGTG CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC GTGCACCCAC AGTTCTACA<mark class="bgy">C</mark></span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta3</td><td><span class="mono">ATCTGGCC<mark class="bgy">G</mark>C CCTCAGGGTG CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC GTGCACCCAC AGTTCTACA<mark class="bgy">C</mark></span></td>
    </tr>
  </table>
</div>

### Bases 950 -- 1049

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">CATCCAGACC GG<mark class="bgr">G</mark>GCGGACA TCGCCCTGCT GGAGCTGGAG GAGCCCGTGA ACATCTCCAG CCAC<mark class="bgr">A</mark>TCCAC AC<mark class="bgx">G</mark>GTCACGC TGCCCCCTGC CTCGGAGACC</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">C<mark class="bgy">AT</mark>CCAGA<mark class="bgy">CT</mark> GGAGCGGA<mark class="bgy">T</mark>A TCGCCCTGCT GGAGCTGGAG GAGCC<mark class="bgy">C</mark>GTGA AC<mark class="bgy">A</mark>TCTCCAG CC<mark class="bgy">G</mark>CGTCCAC ACGGTCA<mark class="bgy">TG</mark>C TGCCCCCTGC CTC<mark class="bgy">G</mark>GAGACC</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta1</td><td><span class="mono">C<mark class="bgy">GC</mark>CCAGA<mark class="bgy">TC</mark> GGAGCGGA<mark class="bgy">C</mark>A TCGCCCTGCT GGAGCTGGAG GAGCC<mark class="bgy">G</mark>GTGA AC<mark class="bgy">G</mark>TCTCCAG CC<mark class="bgy">A</mark>CGTCCAC ACGGTCA<mark class="bgy">CC</mark>C TGCCCCCTGC CTC<mark class="bgy">A</mark>GAGACC</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta2</td><td><span class="mono">C<mark class="bgy">GC</mark>CCAGA<mark class="bgy">TC</mark> GGAGCGGA<mark class="bgy">C</mark>A TCGCCCTGCT GGAGCTGGAG GAGCC<mark class="bgy">G</mark>GTGA A<mark class="bgx">G</mark><mark class="bgy">G</mark>TCTCCAG CC<mark class="bgy">A</mark>CGTCCAC ACGGTCA<mark class="bgy">CC</mark>C TGCCCCCTGC CTC<mark class="bgy">A</mark>GAGACC</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta3</td><td><span class="mono">C<mark class="bgy">GC</mark>CCAGA<mark class="bgy">TC</mark> GGAGCGGA<mark class="bgy">C</mark>A TCGCCCTGCT GGAGCTGGAG GAGCC<mark class="bgy">G</mark>GTGA AC<mark class="bgy">G</mark>TCTCCAG CC<mark class="bgy">A</mark>CGTCCAC ACGGTCA<mark class="bgy">CC</mark>C TGCCCCCTGC CTC<mark class="bgy">A</mark>GAGACC</span></td>
    </tr>
  </table>
</div>

### Bases 1050 -- 1149

<mark class="bgp">beta3FS location is colored pink</mark> [<span class="mono">&ndash;/C</span>]

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG A<mark class="bgr">C</mark>GTGGACAA T<mark class="bgr">A</mark>ATGGTGGG <mark class="bgr">GTGT</mark>TGGGGA CAG<mark class="bgr">C</mark>GGGAGG <mark class="bgr">CC</mark>GGGCCAGG <mark class="bgr">TGGGCACCAA</mark></span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG ATGTGGACAA TGATGGTGGG &ndash;&ndash;TCTGGGGA CAGTGG<mark class="bgy">G</mark>AGG TGGGGCCAGG GTCTTAGCCA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta1</td><td><span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG ATGTGGACAA TGATGGTGGG &ndash;&ndash;TCTGGGGA CAGTGG<mark class="bgy">&ndash;</mark>AGG TGGGGCCAGG GTCTTAGCCA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta2</td><td><span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG ATGTGGACAA TGATGGTGGG &ndash;&ndash;TCTGGGGA CAGTGG<mark class="bgy">&ndash;</mark>AGG TGGGGCCAGG GTCTTAGCCA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta3</td><td><span class="mono">TT<mark class="bgp">CCCCCC</mark>GG GGATGCCGTG CTGGGTCACT GGCTGGGGCG ATGTGGACAA TGATGGTGGG &ndash;&ndash;TCTGGGGA CAGTGG<mark class="bgy">&ndash;</mark>AGG TGGGGCCAGG GTCTTAGCCA</span></td>
    </tr>
  </table>
</div>

### Bases 1150 -- 1249

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono"><mark class="bgr">GTCACAGCCA</mark> <mark class="bgr">CAGGCCAGTC</mark> <mark class="bgr">CGTGGGGTGA</mark>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<mark class="bgr">CCCAG</mark>C<mark class="bgr">C</mark>G<mark class="bgr">G</mark>C CCC<mark class="bgr">AGAC</mark>C<mark class="bgr">C</mark>G GCT<mark class="bgr">C</mark>CAC<mark class="bgr">G</mark>C<mark class="bgr">C</mark></span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">CAGCCCAGCC CCTGG&ndash;&ndash;&ndash;&ndash;&ndash; &ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash; &ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash; &ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash;&ndash; &ndash;&ndash;&ndash;&ndash;&ndash;&ndash;GCTC CAGGTGGGGG TTGCCCGGCC CCCTCCT<mark class="bgy">C</mark>AG GCTGCACCCT</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">CAGCCCAGCC CCTGGCCAGG GTCTTAGCCA CAGCCCAGCC CCTGGGCTCC CTCTGGGCTC CAGGTGGGGG TTGCCCGGCC CCCTCCT<mark class="bgy">G</mark>AG GCTGCACCCT</span></td>
    </tr>
  </table>
</div>

### Bases 1250 -- 1349

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">C<mark class="bgr">CC</mark>C<mark class="bgr">T</mark>CC<mark class="bgr">G</mark>CC <mark class="bgr">CC</mark>CAG<mark class="bgr">T</mark>GC<mark class="bgr">A</mark>C CT<mark class="bgr">G</mark>CC<mark class="bgr">G</mark>CCGC CAT<mark class="bgr">AC</mark>CC<mark class="bgr">G</mark>CT GAAG<mark class="bgr">G</mark>AGGTG <mark class="bgr">G</mark>A<mark class="bgr">A</mark>GTCCCC<mark class="bgr">G</mark> TA<mark class="bgr">G</mark>TGGAAAA CCAC<mark class="bgr">C</mark>TTTG<mark class="bgr">C</mark> <mark class="bgr">A</mark>ACGC<mark class="bgr">GG</mark>AAT A<mark class="bgr">T</mark>CAC<mark class="bgr">ACC</mark>GG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">CTTCCCCACC TGCAGAGC<mark class="bgy">C</mark>C CTC<mark class="bgx">C</mark>CACCGC CATTTCC<mark class="bgy">C</mark>CT GAAGCAGGTG AAGGTCCCCA TAATGGAAAA CCACATTTGT GACGCAAAAT ACCACCTTGG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">CTTCCCCACC TGCAGAGC<mark class="bgy">G</mark>C CTCCCACCGC CATTTCC<mark class="bgy">T</mark>CT GAAGCAGGTG AAGGTCCCCA TAATGGAAAA CCACATTTGT GACGCAAAAT ACCACCTTGG</span></td>
    </tr>
  </table>
</div>

### Bases 1350 -- 1449

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">C<mark class="bgr">CT</mark>C<mark class="bgr">C</mark>A<mark class="bgr">T</mark>ACG GG<mark class="bgr">CC</mark>AC<mark class="bgr">AG</mark>C<mark class="bgr">T</mark> T<mark class="bgr">T</mark>C<mark class="bgr">AA</mark>ATCGT CCG<mark class="bgr">C</mark>GA<mark class="bgr">T</mark>GAC ATGCTGTGTG C<mark class="bgr">G</mark>GGGA<mark class="bgr">G</mark>CG<mark class="bgr">A</mark> <mark class="bgr">AA</mark>A<mark class="bgr">TCA</mark>CGAC</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha1</td><td><span class="mono">CGCCTACACG GGAGACGACG TCCGCATC<mark class="bgy">A</mark>T CCGTGACGAC ATGCTGTGTG CCGGGAACA<mark class="bgy">G</mark> CC<mark class="bgy">A</mark>GAGGGAC TC<mark class="bgy">C</mark>TGC<mark class="bgy">A</mark>AGG TGGGCCCCGC GTGTCCCCCG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha2</td><td><span class="mono">CGCCTACACG GGAGACGACG TCCGCATC<mark class="bgy">A</mark>T CCGTGACGAC ATGCTGTGTG CCGGGAACA<mark class="bgy">C</mark> CC<mark class="bgy">G</mark>GAGGGAC TC<mark class="bgy">A</mark>TGC<mark class="bgy">C</mark>AGG TGGGCCCCGC GTGT<mark class="bgx">C</mark>CCCCG</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">CGCCTACACG GGAGACGACG TCCGCATC<mark class="bgy">G</mark>T CCGTGACGAC ATGCTGTGTG CCGGGAACAC CCGGAGGGAC TC<mark class="bgy">A</mark>TGC<mark class="bgy">C</mark>AGG TGGGCCCCGC GTGTCCCCCG</span></td>
    </tr>
  </table>
</div>

### Bases 1450 -- 1549

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono"><mark class="bgr">GAGCGCATCC</mark> <mark class="bgr">CTCATCCTGA</mark> <mark class="bgr">CCCCCGAAGC</mark> <mark class="bgr">CTGGCCA</mark>GCG AGC<mark class="bgr">A</mark>CTGACC TCTGACCTTC CCAGGG<mark class="bgr">T</mark>GAC TCTGGAGGGC CCCTGGT<mark class="bgr">C</mark>TG CAAGGTGAAT</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha1</td><td><span class="mono">CCCCCCGC<mark class="bgy">A</mark>C CCC<mark class="bgx">A</mark>A<mark class="bgx">T</mark>CCCC ACTCCC<mark class="bgy">G</mark>GG<mark class="bgy">T</mark> CTG<mark class="bgy">G</mark>TCGG<mark class="bgy">T</mark>G AGCGCTGACC TCTGACCTTC CCAGGGCGAC TC<mark class="bgy">T</mark>GGAGGGC CCCTGGTGTG CAAGGTGAAT</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha2</td><td><span class="mono">CCCCCCGC<mark class="bgy">A</mark>C CCCAA<mark class="bgy">C</mark>CCCC ACTCCC<mark class="bgy">A</mark>GG<mark class="bgy">C</mark> CTG<mark class="bgy">T</mark>TCGG<mark class="bgy">C</mark>G AGCGCTGACC TCTGACCTTC CCAGGGCGAC TC<mark class="bgy">T</mark>GGAGGGC CCCTGGTGTG CAAGGTGAAT</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">CCCCCCGC<mark class="bgx">C</mark>C CCCAACCCCC ACTCCCAGGC CTG<mark class="bgx">T</mark>TCGGCG AGCGCTGACC TCTGACCTTC CCAGGGCGAC TC<mark class="bgy">C</mark>GGAGGGC CCCTGGTGTG CAAGGTGAAT</span></td>
    </tr>
  </table>
</div>

### Bases 1550 -- 1649

Alignment near putative functional alpha/beta mutation in <mark class="bgg">green</mark> (cf. `r AutoCite(maun)` supplement)

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">GGCACCT<mark class="bgr">AA</mark>C TGCAGGCGGG CGTGGTCAGC TGGGA<mark class="bgr">G</mark>GAG<mark class="bgr">A</mark> GCTGTGCCCA GCCCAACCGG CCTGGCATCT ACACCCGTGT CACCTACTAC TTGGACTGGA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">GGCACCTGGC T<mark class="bgy">A</mark>CAGGCGGG CGTGGTCAGC TGGG<mark class="bgg">A</mark>CGAGG GCTGTGCCCA GCCCAACCGG CCTGGCATCT ACACCCGTGT CACCTACTAC TTGGACTGGA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">GGCACCTGGC T<mark class="bgy">G</mark>CAGGCGGG CGTGGTCAGC TGGG<mark class="bgg">G</mark>CGAGG GCTGTGCCCA GCCCAACCGG CCTGGCATCT ACACCCGTGT CACCTACTAC TTGGACTGGA</span></td>
    </tr>
  </table>
</div>

### Bases 1650 -- 1749

<div class="evensuperwider">
  <table class="table table-condensed">
    <tr style="border-top: 1px solid #ccc;">
      <td width="8%;">delta</td><td><span class="mono">TCCACCACTA TGTCCCCAA<mark class="bgr">G</mark> AAGCC<mark class="bgr">C</mark>TGAG <mark class="bgr">C</mark>CAGGCCTGG GGTGTCCACC <mark class="bgr">C</mark>GGGTCACTG GAGGGCCAGC CCCTCCTGTC CAAA<mark class="bgr">C</mark>CACCA CTGCTTCCTA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>alpha</td><td><span class="mono">TCCACCACTA TGTCCCCAAA AAGCCGTGAG TCAGGCCTGG GTTGGCCACC TGGGTCACTG GAG<mark class="bgx">GA</mark>CCA<mark class="bgx">AC</mark> CCCT<mark class="bgx">G</mark>CTGTC CAAAACACCA CTGCTTCCTA</span></td>
    </tr>
    <tr style="border-top: 1px solid #ccc;">
      <td>beta</td><td><span class="mono">TCCACCACTA TGTCCCCAAA AAGCCGTGAG TCAGGCCTGG G<mark class="bgx">T</mark>TG<mark class="bgx">G</mark>CCACC TGGGTCACTG GAG<mark class="bgx">GA</mark>CCA<mark class="bgx">AC</mark> CCCT<mark class="bgx">G</mark>CTGTC CAAAACACCA CTGCTTCCTA</span></td>
    </tr>
  </table>
</div>

## Citing `TrypLik`

We hope that `r Biocpkg("TrypLik")` will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!

```{r "citation"}
## Citation info
citation("TrypLik")
```
# Reproducibility

The `r Biocpkg("TrypLik")` package `r Citep(bib[["TrypLik"]])` was made possible thanks to:

* R `r Citep(bib[["R"]])`
* `r Biocpkg("BiocStyle")` `r Citep(bib[["BiocStyle"]])`
* `r Biocpkg("biocthis")` `r Citep(bib[["biocthis"]])`
* `r CRANpkg("knitr")` `r Citep(bib[["knitr"]])`
* `r CRANpkg("RefManageR")` `r Citep(bib[["RefManageR"]])`
* `r CRANpkg("rmarkdown")` `r Citep(bib[["rmarkdown"]])`


# Bibliography

This vignette was generated using `r Biocpkg("BiocStyle")` `r Citep(bib[["BiocStyle"]])`
with `r CRANpkg("knitr")` `r Citep(bib[["knitr"]])` and `r CRANpkg("rmarkdown")` `r Citep(bib[["rmarkdown"]])` running behind the scenes.

Citations made with `r CRANpkg("RefManageR")` `r Citep(bib[["RefManageR"]])`.

```{r vignetteBiblio, results = "asis", echo = FALSE, warning = FALSE, message = FALSE}
## Print bibliography
PrintBibliography(bib, .opts = list(hyperlink = "to.doc", style = "html", max.names = 3))
```
