<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="TrypLik">
<title>Estimating Tryptase Copy Numbers from Sequence Data • TrypLik</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Estimating Tryptase Copy Numbers from Sequence Data">
<meta property="og:description" content="TrypLik">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TrypLik</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/TrypLik.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Estimating Tryptase Copy Numbers from Sequence Data</h1>
                        <h4 data-toc-skip class="author">Jeff Wall</h4>
            <address class="author_afil">
      Oregon Health and Science
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:jeffwall.genetics@gmail.com" class="email">jeffwall.genetics@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Jonathan
Carroll</h4>
            <address class="author_afil">
      Human Immunology
Biosciences<br><a class="author_email" href="mailto:#"></a><a href="mailto:jonathan@hibio.com" class="email">jonathan@hibio.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">2 May 2024</h4>
      
      
      <div class="d-none name"><code>TrypLik.Rmd</code></div>
    </div>

    
    
<!-- # Basics -->
<!-- ## Install `TrypLik` R package -->
<!-- The R package can be installed with the following: -->
<!-- ```{r "install", eval = FALSE} -->
<!-- if (!requireNamespace("BiocManager", quietly = TRUE)) { -->
<!--       install.packages("BiocManager") -->
<!--   } -->
<!-- BiocManager::install("TrypLik") -->
<!-- ## Check that you have a valid Bioconductor installation -->
<!-- BiocManager::valid() -->
<!-- ``` -->
<!-- # Command-line usage -->
<!-- Additionally, the C code which performs the internal aggregation is provided in  -->
<!-- `inst/tryplik.c` for use within a pipeline outside of R. -->
<style type="text/css">
.bgr {
  background-color: salmon;
  padding: 0;
}
.bgy {
  background-color: yellow;
  padding: 0;
}
.bgg {
  background-color: springgreen;
  padding: 0;
}
.bgp {
  background-color: pink;
  padding: 0;
}
.tg {
  background-color: unset;
  font-weight: bold;
  padding: unset;
  color: green
}
.bgx {
  background-color: silver;
  padding: 0;
}
.mono {
  font-family: monospace;
  font-size: 120%;
}
.lefttable table {
  margin-left: 50px;
  max-width: 700px;
}
.superwide table {
  margin-left: 50px;
  max-width: 1000px;
  width: 120%;
}
.evensuperwider table {
  margin-left: 50px;
  max-width: 1600px;
  width: 150%;
}
</style>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>We have developed a likelihood-based approach for estimating tryptase
copy numbers (specifically the numbers of <span class="math inline">\(\alpha\)</span> alleles, <span class="math inline">\(\beta\)</span> alleles and <span class="math inline">\(\beta\)</span> frameshift alleles) directly from
whole-genome (preferable) or whole-exome sequence data. The approach
utilizes an updated assembly of canonical <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, and <span class="math inline">\(\delta\)</span> tryptase alleles, and identifies
reads that map uniquely to only one of these alleles. The complete
workflow is implemented in the package TrypLik.</p>
</div>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>The tryptase gene cluster contains four closely spaced paralogous
genes on chromosome 16p13.3. These include TPSAB1 and TPSB2, which
together encode the <span class="math inline">\(\alpha\)</span> and
<span class="math inline">\(\beta\)</span> tryptase peptides. TPSAB1
encodes two major alleles, <span class="math inline">\(\alpha\)</span>
and <span class="math inline">\(\beta\)</span>I, while TPSB2’s major
alleles are <span class="math inline">\(\beta\)</span>II and <span class="math inline">\(\beta\)</span>III. <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span>I are <span class="math inline">\(\sim1\%\)</span> diverged over a <span class="math inline">\(1.2\)</span> Kbp region, while <span class="math inline">\(\beta\)</span>I and <span class="math inline">\(\beta\)</span>II differ by just a single
nucleotide over the same region. <span class="math inline">\(\alpha\)</span> alleles are thought to be
enzymatically inactive, and while most <span class="math inline">\(\beta\)</span> alleles are active, there is a
specific <span class="math inline">\(\beta\)</span> frameshift variant
(<span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span>) that is inactive.</p>
<p>Recent work has identified a familial inherited syndrome, hereditary
alpha tryptasemia (H<span class="math inline">\(\alpha\)</span>T), which
is associated with one or more duplications of the <span class="math inline">\(\alpha\)</span> allele in TPSAB1 Lyons et al.,
2016. Common phenotypes found in H<span class="math inline">\(\alpha\)</span>T patients include systemic venom
reactions, pruritis, IBS, reflux, skeletal abnormalities, retained
primary dentition and anthralgia. There is also evidence for <span class="math inline">\(\beta\)</span> duplications and deletions (Chollet
et al., 2021, Glover et al., 2022), though it is not yet clear if this
reflects copy number variants (CNVs) in only TPSAB1 or CNVs in both
TPSAB1 and TPSAB2. Nothing is known about the clinical effects of
tryptase <span class="math inline">\(\beta\)</span> copy number
variation.</p>
<p>While <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, and <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> copy numbers can be accurately
determined using custom droplet digital PCR (ddPCR) assays (cf. Lyons et
al., 2016), it is much more challenging to estimate copy numbers from
standard short-read (e.g. Illumina) sequence data, even with a high
depth of coverage. This is primarily due to the fact that <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, and <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> tryptase alleles are so similar,
but exacerbated by errors in the human reference genome assembly (hg38),
which appears to have chimeric alleles for both TPSAB1 and TPSB2. In
addition, the published allelic sequences in Maun et al., 2019 also
appear to be incorrect. This has motivated our efforts to generate our
own consensus assemblies of different tryptase alleles (see below).</p>
</div>
<div class="section level2">
<h2 id="instructions-on-use">Instructions on use<a class="anchor" aria-label="anchor" href="#instructions-on-use"></a>
</h2>
<p>There are two main parts to the workflow: First, we map reads to a
tryptase consensus sequence and count the ones that map uniquely to
tryptase <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> and <span class="math inline">\(\delta\)</span> alleles. Second, we perform
Bayesian likelihood calculations to estimate the most likely copy number
count. We describe each step in greater detail below.</p>
<div class="section level3">
<h3 id="tryptase-read-mapping">Tryptase read mapping<a class="anchor" aria-label="anchor" href="#tryptase-read-mapping"></a>
</h3>
<p>This part of the workflow can be broken up into the following
components:</p>
<ol style="list-style-type: decimal">
<li>Subset an indexed bam file to consider only reads mapping to the
tryptase region</li>
<li>Convert the output back to (the original) paired-end fastq
reads</li>
<li>Map raw reads to consensus tryptase sequence</li>
<li>Remove duplicates</li>
<li>Count the numbers of reads containing specific 30 bp motifs that map
specifically to a single tryptase allele</li>
</ol>
<p>There are different ways to go about each of these steps:</p>
<div class="section level4">
<h4 id="indexing-bam-files">Indexing bam files<a class="anchor" aria-label="anchor" href="#indexing-bam-files"></a>
</h4>
<p>If a datafile is not already indexed, we recommend using samtools to
do so. For example, we use publicly available high-coverage whole genome
sequence data from the 1000 Genomes Project as a test case. We
downloaded the file “HG00100.final.cram” from <a href="https://www.internationalgenome.org/data-portal/sample/HG00100" class="external-link uri">https://www.internationalgenome.org/data-portal/sample/HG00100</a>
to use as an example.Indexing and subsetting can be performed with the
following commands:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">samtools</span> index HG00100.final.cram</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-b</span> <span class="at">-h</span> HG00100.final.cram chr16:1200000-1300000 <span class="op">&gt;</span> HG00100.tryptase.cram</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">samtools</span> index HG00100.tryptase.cram</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="converting-indexed-cram-files-to-fastq-reads">Converting indexed cram files to fastq reads<a class="anchor" aria-label="anchor" href="#converting-indexed-cram-files-to-fastq-reads"></a>
</h4>
<p>One possibility for doing this uses samtools again:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="at">-n</span> HG00100.tryptase.cram <span class="at">-o</span> HG00100.tryptase.sorted.cram</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">samtools</span> fastq <span class="at">-1</span> HG00100.tryptase.R1.fastq <span class="at">-2</span> HG00100.tryptase.R2.fastq <span class="at">-0</span> /dev/null <span class="at">-s</span> /dev/null <span class="at">-n</span> HG00100.tryptase.sorted.cram</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="mapping-reads-to-tryptase-consensus-sequence">Mapping reads to tryptase consensus sequence<a class="anchor" aria-label="anchor" href="#mapping-reads-to-tryptase-consensus-sequence"></a>
</h4>
<p>Here, we are using the tryptase consensus sequence published by Maun
et al., 2019 to winnow out reads that are in the tryptase region that do
not actually map to tryptase genes. The following command works for our
continuing example:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">bwa</span> mem consensus.fa HG00100.tryptase.R1.fastq HG00100.tryptase.R2.fastq <span class="op">&gt;</span> out.sam</span></code></pre></div>
<p>The file <code>consensus.fa</code> is included with this package in
<a href="inst/consensus.fa">inst/consensus.fa</a>.</p>
</div>
<div class="section level4">
<h4 id="removing-duplicates">Removing duplicates<a class="anchor" aria-label="anchor" href="#removing-duplicates"></a>
</h4>
<p>This is generally done using Picard or samtools. Here is an example
of doing so using samtools, with the output written to the file
“temp.sam”.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="at">-n</span> out.sam <span class="at">-o</span> out.sorted.sam</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">samtools</span> fixmate <span class="at">-m</span> out.sorted.sam fixmate.sam</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="at">-o</span> positionsort.sam fixmate.sam</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">samtools</span> markdup <span class="at">-r</span> positionsort.sam temp.sam</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="counting-reads-mapping-to-specific-tryptase-alleles">Counting reads mapping to specific tryptase alleles<a class="anchor" aria-label="anchor" href="#counting-reads-mapping-to-specific-tryptase-alleles"></a>
</h4>
<p>The file “count_tryptase.sh” is a shell script that counts the
numbers of reads that map specifically to different tryptase alleles
(see below for technical details). It requires the input file to be
called “temp.sam”. Output consists of <span class="math inline">\(9\)</span> integers summarizing read counts:</p>
<ul>
<li><p>The first four of these are the numbers of reads mapping uniquely
to <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>I, <span class="math inline">\(\beta\)</span>II or <span class="math inline">\(\beta\)</span>III alleles in the specific <span class="math inline">\(83\)</span> bp region where all four alleles can
be distinguished from each other.</p></li>
<li><p>The next two are the numbers of reads that map uniquely to <span class="math inline">\(\beta\)</span> ‘wildtype’ (i.e. not frameshifted)
and <span class="math inline">\(\beta\)</span> frameshifted
sequences</p></li>
<li><p>The final three are the numbers of reads that can be uniquely
mapped to <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, or <span class="math inline">\(\delta\)</span> alleles respectively</p></li>
</ul>
<p>Further details on these can be found in the technical details below.
For our example with sample HG00100 from the 1000 Genomes Project, the
output of “count_tryptase” is</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">24</span>    0   22    0   26    0   347   316   304</span></code></pre></div>
<p>The above processing can be performed in R (provided samtools and bwa
are available) with the following wrappers</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="st">"HG00100.final.cram"</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/count_tryptase.html">count_tryptase</a></span><span class="op">(</span><span class="va">sam_file</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span></span>
<span><span class="co">#&gt; [1]  24   0  22   0  26   0 347 316 304</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="tryptase-copy-number-likelihood-calculations">Tryptase copy number likelihood calculations<a class="anchor" aria-label="anchor" href="#tryptase-copy-number-likelihood-calculations"></a>
</h3>
<p>The last five numbers from the output of “count_tryptase” are used as
input to the program TrypLik to estimate the likelihoods of different
combinations of total number of <span class="math inline">\(\alpha\)</span> alleles, total number of <span class="math inline">\(\beta\)</span> alleles, and total number of <span class="math inline">\(\beta\)</span> frameshift alleles. The method
accounts for dependence across read counts (caused by the same <span class="math inline">\(2 \times 150\)</span> bp read generally
overlapping multiple of the regions used to distinguish between alleles)
and also uses prior information on CNV frequencies in different
populations. Usage is</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">TrypLik</span> WT FS Areads Breads Dreads <span class="er">(</span><span class="ex">-pop</span> POP<span class="kw">)</span></span></code></pre></div>
<p><code>WT</code>: number of reads mapping to <span class="math inline">\(\beta\)</span> “wildtype” alleles at the site of
the frameshift mutation (5th field from “count_tryptase” output)</p>
<p><code>FS</code>: number of reads mapping to <span class="math inline">\(\beta\)</span> frameshift alleles at the site of
the frameshift mutation (6th field from “count_tryptase” output)</p>
<p><code>Areads</code>: number of reads mapping uniquely to <span class="math inline">\(\alpha\)</span> alleles relative to <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\delta\)</span> alleles (7th field from
“count_tryptase” output)</p>
<p><code>Breads</code>: number of reads mapping uniquely to <span class="math inline">\(\beta\)</span> alleles relative to <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\delta\)</span> alleles (8th field from
“count_tryptase” output)</p>
<p><code>Dreads</code>: number of reads mapping uniquely to <span class="math inline">\(\delta\)</span> alleles relative to <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> alleles (9th field from
“count_tryptase” output)</p>
<p><code>-pop POP</code>: optional flag to specify the predominant
ancestry of the test sample. Options include EUR for European ancestry,
SAS for South Asian ancestry, EAS for East Asian ancestry and AFR for
African ancestry. The default, if the <code>-pop</code> flag is not
included, is European ancestry.</p>
<p>Note that these likelihood calculations assume that all individuals
contain exactly two tryptase <span class="math inline">\(\delta\)</span>
alleles. <code>Dreads</code> acts as a ‘control’ from which we estimate
the other copy numbers. The output consists of a list of different <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\beta\)</span> frameshift allele copy numbers
followed by the combination’s posterior probability. To continue our
example from before, calculation for HG00100 would use the command</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">TrypLik</span> 26 0 347 316 304 <span class="at">-pop</span> EUR</span></code></pre></div>
<p>Output would look like this:</p>
<table class="table">
<thead><tr class="header">
<th align="left">Alpha count</th>
<th align="left">Beta count</th>
<th align="left">Beta FS count</th>
<th align="left">Posterior likelihood</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="left">2</td>
<td align="left">0</td>
<td align="left">0.99890</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">3</td>
<td align="left">0</td>
<td align="left">0.00008</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">2</td>
<td align="left">0</td>
<td align="left">0.00082</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">3</td>
<td align="left">0</td>
<td align="left">0.00020</td>
</tr>
</tbody>
</table>
<p>The method estimates that HG00100 has a 2<span class="math inline">\(\alpha:2\beta:2\delta\)</span> genotype (with no
frameshift alleles) with posterior probability 99.89%.</p>
<p>An R wrapper for this function is provided, which takes either a
single vector of values or individual arguments</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TrypLik.html">TrypLik</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Alpha_count Beta_count Beta_FS_count Posterior_likelihood</span></span>
<span><span class="co">#&gt; 1           2          2             0              0.99890</span></span>
<span><span class="co">#&gt; 2           2          3             0              0.00008</span></span>
<span><span class="co">#&gt; 3           3          2             0              0.00082</span></span>
<span><span class="co">#&gt; 4           3          3             0              0.00020</span></span>
<span></span>
<span><span class="co"># or TrypLik(26, 0, 347, 316, 304, pop = 0)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="technical-details">Technical details<a class="anchor" aria-label="anchor" href="#technical-details"></a>
</h2>
<div class="section level3">
<h3 id="de-novo-assemblyalignment">de novo assembly/alignment<a class="anchor" aria-label="anchor" href="#de-novo-assemblyalignment"></a>
</h3>
<p>We started by aligning cDNA sequences to a 40 bp region near the
probes used by Lyons et al., 2016(Supplementary Figure 3b) for
ddPCR.</p>
<div class="lefttable">
<table class="table">
<colgroup>
<col width="33%">
<col width="66%">
</colgroup>
<tbody>
<tr class="odd">
<td>Delta (CCDS10432.1)</td>
<td align="left"><span class="mono">CCAGGCCAGG CCCTGCAGCA
A<mark class="bgr">A</mark>CGGGCAT<mark class="bgr">T</mark>
GTTGGGGG<mark class="bgr">G</mark>C</span></td>
</tr>
<tr class="even">
<td>Alpha (M30038.1)</td>
<td align="left"><span class="mono">CCAG<mark class="bgy">T</mark>CCAGG
CCCTGCAGC<mark class="bgy">A</mark>
AG<mark class="bgy">C</mark>GGG<mark class="bgy">T</mark>ATC
GT<mark class="bgy">C</mark>GGGGGTC</span></td>
</tr>
<tr class="odd">
<td>Beta1 (M33492.1)</td>
<td align="left"><span class="mono">CCAG<mark class="bgy">G</mark>CCAGG
CCCTGCAGC<mark class="bgy">G</mark>
AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC
GT<mark class="bgy">C</mark>GGGGGTC</span></td>
</tr>
<tr class="even">
<td>Beta2 (BC065923.1)</td>
<td align="left"><span class="mono">CCAG<mark class="bgy">G</mark>CCAGG
CCCTGCAGC<mark class="bgy">G</mark>
AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC
GT<mark class="bgy">T</mark>GGGGGTC</span></td>
</tr>
<tr class="odd">
<td>Beta3 (AF099143)</td>
<td align="left"><span class="mono">CCAG<mark class="bgy">G</mark>CCAGG
CCCTGCAGC<mark class="bgy">G</mark>
AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC
GT<mark class="bgy">T</mark>GGGGGTC</span></td>
</tr>
</tbody>
</table>
</div>
<p>Here sites in red are fixed differences between TPSD1 and
(TPSAB1+TPSB2), while sites in yellow can be used to distinguish <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>I, and <span class="math inline">\(\beta\)</span>II/<span class="math inline">\(\beta\)</span>III alleles from each other. <span class="math inline">\(\beta\)</span>II and <span class="math inline">\(\beta\)</span>III alleles are distinguishable from
each other at sites near the Lyons et al., 2016 REV primer as
follows:</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="66%">
</colgroup>
<tbody>
<tr class="odd">
<td>Beta2 (BC065923.1)</td>
<td align="left"><span class="mono">GAGAGTCC<mark class="bgy">A</mark>C
G<mark class="bgy">G</mark>CC<mark class="bgy">C</mark>ATACT</span></td>
</tr>
<tr class="even">
<td>Beta3 (AF099143)</td>
<td align="left"><span class="mono">GAGAGTCC<mark class="bgy">G</mark>C
G<mark class="bgy">A</mark>CC<mark class="bgy">G</mark>ATACT</span></td>
</tr>
</tbody>
</table>
<p>We assumed that the sequences listed above are correct and invariant
in the different tryptase alleles. We then used Illumina short read data
to ‘extend’ allele-specific alignments. Specifically, we combined the
raw reads from 30 randomly chosen individuals of European ancestry from
the 1000 Genomes Project (high-coverage data generated by the NYGC) and
pulled out those reads that matched the above sequences. We then could
expand on the allele-specific assemblies by looking at the alignments
just 5’ and 3’ to the sequences shown above, and concurrently identify
common allele-specific polymorphisms. We continued to iteratively extend
these assemblies until <span class="math inline">\(\alpha\)</span> and
<span class="math inline">\(\beta\)</span> alleles could not be
distinguished from each other. Roughly speaking, since the Illumina data
consisted of 150 bp reads, this process can continue as long as there is
a fixed difference between allelic assemblies every <span class="math inline">\(\sim 150\)</span> bp. We extended these assemblies
across <span class="math inline">\(\sim 1.1\)</span> Kbp of sequence
including most of the coding regions (see Appendix 1).</p>
<p>Our alignments show the expected difference between <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> alleles at the putative functional
mutation. In addition, we confirm that the frameshift mutation happens
only on a <span class="math inline">\(\beta\)</span>III background
(leading to the inactivated <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> allele), and that the <span class="math inline">\(\beta\)</span>I and <span class="math inline">\(\beta\)</span>II consensus sequences differ by
just a single nucleotide, despite <span class="math inline">\(\beta\)</span>I being an allele that is (supposed
to be) exclusive to TPSAB1 and <span class="math inline">\(\beta\)</span>II being confined to TPSB2. This
likely reflects paralogous gene conversion between tryptase genes over
evolutionarily recent time scales.</p>
<p>The tryptase allele assemblies generated here differ slightly from
all previously published assemblies, including some of the cDNA
sequences, the Lyons et al., 2016 tryptase alignment, the Maun et al.,
2019 tryptase alignment, and the human reference genome (hg38). We
believe though that Illumina reads are highly accurate, and can show how
these previous assemblies are inconsistent with Illumina data from
multiple individuals.</p>
</div>
</div>
<div class="section level2">
<h2 id="tryptase-cnv-genotyping-from-high-coverage-illumina-reads">Tryptase CNV genotyping from high-coverage Illumina reads<a class="anchor" aria-label="anchor" href="#tryptase-cnv-genotyping-from-high-coverage-illumina-reads"></a>
</h2>
<p>From the alignment described above, we identified <span class="math inline">\(11\)</span> different <span class="math inline">\(30\)</span> bp regions where reads could be
uniquely mapped to <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, or <span class="math inline">\(\delta\)</span> alleles. Using the base numbering
defined by Maun et al., 2019(Data S1) and also used in the accompanying
table, these correspond to the following regions:</p>
<span class="math display">\[\begin{array}{ccc}
560\, – 589 &amp; 1010\, – 1039 &amp; 1360\, – 1389 \\
740\, – 769 &amp; 1115 \sim 1146 &amp; 1510\, – 1539 \\
840\, – 869 &amp; 1220\, – 1249 &amp; 1560\, – 1589 \\
940\, – 969 &amp; 1260\, – 1289
\end{array}\]</span>
<p>Tallying up the total number of ‘<span class="math inline">\(\alpha\)</span> reads’, ‘<span class="math inline">\(\beta\)</span> reads’ and ‘<span class="math inline">\(\delta\)</span> reads’ (which we call here <span class="math inline">\(A\)</span>, <span class="math inline">\(B\)</span>
and <span class="math inline">\(D\)</span> respectively) gives us a way
of estimating the relative copy numbers of <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\delta\)</span> alleles. In particular, if we
assume each individual has exactly two copies of TPSD1 (i.e. no copy
number variation in TPSD1), then the number of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> alleles can be estimated by <span class="math inline">\(2A/D\)</span> and <span class="math inline">\(2B/D\)</span> respectively.</p>
<p>Some caveats apply:</p>
<ol style="list-style-type: decimal">
<li><p>The regions are not independent of each other, since the <span class="math inline">\(2 \times 150\)</span> bp paired-end reads
generally include multiple regions. This leads to a greater variance
than otherwise expected in estimated allele counts.</p></li>
<li><p>Even in a pool of individuals with the same true number of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> alleles, the estimates <span class="math inline">\(2A/D\)</span> and <span class="math inline">\(2B/D\)</span> will be correlated since they both
have the same denominator.</p></li>
<li><p>Unknown (or unaccounted for) polymorphisms within an allelic
class can lead to misclassified reads. This is perhaps most important in
the analysis of non-European ancestry individuals.</p></li>
<li><p>The method assumes that all CNVs affect the entire <span class="math inline">\(1\)</span> Kbp test region, though we know this
isn’t true. Chovanec, 2022 report the existence of σ alleles, which
appear to be a chimera of <span class="math inline">\(\alpha\)</span>
and <span class="math inline">\(\beta\)</span> alleles. In our analyses,
these alleles are estimated as partial <span class="math inline">\(\alpha\)</span> and partial <span class="math inline">\(\beta\)</span>, proportional to the numbers of
reads mapping to each canonical allele assembly.</p></li>
</ol>
<p>We suggest using an empirically-based approach for estimating the
bias in the estimates (e.g. estimating</p>
<p><span class="math display">\[
E \bigl( \frac{2A}{D} |\ \textrm{actual number of }\alpha\textrm{
alleles is known} \bigr)
\]</span></p>
<p>that uses orthogonal information on true allele counts
(e.g. validated ddPCR data). A preliminary version of this is described
below.</p>
</div>
<div class="section level2">
<h2 id="betaiii-frameshift-mutation">
<span class="math inline">\(\beta\)</span>III frameshift
mutation<a class="anchor" aria-label="anchor" href="#betaiii-frameshift-mutation"></a>
</h2>
<p>The approach described above can estimate the total copy number of
tryptase beta alleles, but some of these beta alleles are inactive due
to a frameshift mutation. In parallel, we conduct a read-counting
exercise that counts reads that uniquely map to the <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> allele and reads that map to all
other <span class="math inline">\(\beta\)</span> alleles:</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="66%">
</colgroup>
<tbody>
<tr class="odd">
<td>
<span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span>
</td>
<td align="left"><span class="mono">CTCAGAGACC TTCCCCCCCG
GGGATGCCGT</span></td>
</tr>
<tr class="even">
<td>All other <span class="math inline">\(\beta\)</span>
</td>
<td align="left"><span class="mono">CTCAGAGACC TTCCCCCC–G
GGGATGCCGTG</span></td>
</tr>
</tbody>
</table>
<p>(Note that both <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\sigma\)</span> alleles do not map to either of
these sequences.) The observed numbers of reads in each of these two
categories (called here FS+ and FS-) is expected to follow a binomial
distribution with probabilities proportional to the true numbers of
<span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> and non-<span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> <span class="math inline">\(\beta\)</span> alleles. Further, by assumption,
<span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> alleles only occur in TPSB2. If
data = {FS+, FS-}, then under a simple likelihood framework,</p>
<p><span class="math display">\[
\rm{lik} \bigl(data\bigr) \sim Pr \bigl(genotype\bigr)\ Pr \bigl( data\
|\ genotype\bigr)
\]</span></p>
<p>Suppose for example we observe <span class="math inline">\(28\)</span> and <span class="math inline">\(36\)</span> reads supporting <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> and non-<span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> <span class="math inline">\(\beta\)</span> alleles respectively (i.e. data =
<span class="math inline">\(\{28, 36\}\)</span>) in an individual of
European ancestry with an estimated tryptase CNV genotype of <span class="math inline">\(0\alpha:4\beta:2\delta\)</span>. Based on the
European <span class="math inline">\(1\)</span> KGP data, the frequency
of the <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> mutation is <span class="math inline">\(\sim\!0.1766\)</span>. So, the naïve (i.e. prior
or Hardy-Weinberg) probability of a homozygous <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> genotype (i.e. <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> mutations on both chromosomes) is
<span class="math inline">\((0.1766)^2\)</span> and the probability of
having exactly one copy of <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> is <span class="math inline">\(2(0.1766)(1-0.1766)\)</span>. Assuming the
estimated tryptase CNV genotype is correct, we calculate</p>
<p><span class="math display">\[
\textrm{Pr} \bigl(\textrm{data}\ |\ \beta\textrm{III}^{FS}\,
\textrm{heterozygote}\bigr) = {64 \choose 28} (0.25)^{28}(0.75)^{36},\
\textrm{and}
\]</span> <span class="math display">\[
\textrm{Pr} \bigl(\textrm{data}\ |\ \beta\textrm{III}^{FS}\,
\textrm{homozygote}\bigr) = {64 \choose 28} (0.5)^{28} (0.5)^{36}
\]</span></p>
<p>and</p>
<p><span class="math display">\[
\textrm{lik} \bigl(\beta\textrm{III}^{FS}\, \textrm{homozygote}\bigr)
\sim 0.0018922,\ \textrm{and}
\]</span> <span class="math display">\[
\textrm{lik} \bigl(\beta\textrm{III}^{FS}\, \textrm{heterozygote}\bigr)
\sim 0.00014354
\]</span></p>
<p>In a Bayesian context, the posterior <span class="math inline">\(\textrm{Pr} \bigl(\beta\textrm{III}^{FS}\,
\textrm{homozygote}\bigr) = 0.9295\)</span>, and we conclude that the
individual is likely to have <span class="math inline">\(0
\alpha\)</span> alleles, <span class="math inline">\(2\)</span> active
<span class="math inline">\(\beta\)</span> alleles and <span class="math inline">\(2\)</span> inactive <span class="math inline">\(\beta\)</span>III<span class="math inline">\(^{FS}\)</span> alleles.</p>
</div>
<div class="section level2">
<h2 id="tryptase-cnv-genotyping-revisited-likelihood-framework">Tryptase CNV genotyping revisited: likelihood framework<a class="anchor" aria-label="anchor" href="#tryptase-cnv-genotyping-revisited-likelihood-framework"></a>
</h2>
<p>In an idealized world where the 4 caveats described above do not
apply, each read would be independent and the read counts <span class="math inline">\((A, B, D)\)</span> could be modeled as following a
multinomial distribution conditional on the sum <span class="math inline">\(A+B+D\)</span>, with probabilities for each
category proportional to the true copy numbers of <span class="math inline">\(\alpha\)</span> <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\delta\)</span>. We employ two modifications to
this model to account for deviations from model assumptions:</p>
<ol style="list-style-type: decimal">
<li><p>We replace <span class="math inline">\((A+B+D)\)</span> by <span class="math inline">\((A+B+D) / R\)</span>, the “effective number of
independent read counts”, to account for caveat #1</p></li>
<li><p>We reweight the multinomial probabilities by factors of <span class="math inline">\(E(2A/D\ |\ \alpha\ \textrm{copy number} =
1)\)</span> and <span class="math inline">\(E(2B/D\ |\ \beta\
\textrm{copy number} = 3)\ /\ 3\)</span> for <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> copy numbers to account for unknown
structural variants and/or polymorphisms</p></li>
</ol>
<p>We use empirical data from the EUR populations from the 1000 Genomes
Project to estimate these correction factors. First, we plotted <span class="math inline">\(2A/D\)</span> vs. <span class="math inline">\(2B/D\)</span> for these <span class="math inline">\(503\)</span> samples:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">TrypLik</span><span class="fu">::</span><span class="fu"><a href="../reference/tryptase_1KG.html">tryptase_1KG</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">genomes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">`2A/D`</span>, <span class="va">`2B/D`</span>, col <span class="op">=</span> <span class="va">`Geno Group`</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="TrypLik_files/figure-html/genomes-1.png" width="700"></p>
<p>The source for this data is available to an R session with
<code><a href="../reference/tryptase_1KG.html">tryptase_1KG()</a></code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/tryptase_1KG.html">tryptase_1KG</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;   2A/D   2B/D Geno Group</span></span>
<span><span class="co">#&gt; 1    0 2.9885      0α:4β</span></span>
<span><span class="co">#&gt; 2    0 3.2629      0α:4β</span></span>
<span><span class="co">#&gt; 3    0 3.4384      0α:4β</span></span>
<span><span class="co">#&gt; 4    0 3.4791      0α:4β</span></span>
<span><span class="co">#&gt; 5    0 3.4792      0α:4β</span></span>
<span><span class="co">#&gt; 6    0 3.5294      0α:4β</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="../reference/tryptase_1KG.html">tryptase_1KG</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       2A/D   2B/D      Geno Group</span></span>
<span><span class="co">#&gt; 498 3.3934 2.1381 everything else</span></span>
<span><span class="co">#&gt; 499 3.5198 2.6565 everything else</span></span>
<span><span class="co">#&gt; 500 3.6170 2.0479 everything else</span></span>
<span><span class="co">#&gt; 501 3.6826 2.3413 everything else</span></span>
<span><span class="co">#&gt; 502 3.6910 2.1691 everything else</span></span>
<span><span class="co">#&gt; 503 3.9388 1.9878 everything else</span></span></code></pre></div>
<p>We focus on the dots colored green, which are estimated to have a
tryptase genotype of <span class="math inline">\(1\alpha:3\beta:2\delta\)</span>. The average
values of <span class="math inline">\(2A/D\)</span> and <span class="math inline">\(2B/D\)</span> are <span class="math inline">\(1.06\)</span> and <span class="math inline">\(3.13\)</span> respectively, and these means are
relatively insensitive to the inclusion/exclusion of specific points on
the periphery of the central mass. In addition, we note that</p>
<p><span class="math display">\[
\begin{eqnarray}
\textrm{Var}(2A/D) &amp;=&amp; 0.0263 \\
\textrm{Var}(2B/D) &amp;=&amp; 0.1286 \\
r^2 (2A/D, 2B/D) &amp;=&amp; 0.1953 \\
\end{eqnarray}
\]</span></p>
<p>We then ran simulations with different values of R (but conditional
on the observed <span class="math inline">\(A\)</span>, <span class="math inline">\(B\)</span> and <span class="math inline">\(D\)</span> values) to heuristically find a value
that, on average, produces correlation and variances similar to what was
observed. We settled on <span class="math inline">\(R = 2.75\)</span>,
but with the caveat that this estimation process was entirely
qualitative. We do note though that this <span class="math inline">\(R\)</span> is approximately equal to the number of
<span class="math inline">\(30\)</span> bp informative regions covered
by an average <span class="math inline">\(2\times 150\)</span> bp read
pair in the target tryptase region. As an example of the process,
suppose the observed data <span class="math inline">\(\{A, B, D\} =
\{192, 600, 290\}\)</span> and that we want to estimate the probability
of observing the data given a true genotype of <span class="math inline">\(1\alpha:4\beta\)</span>. Then, <span class="math inline">\(2A/D = 1.324, 2B/D = 4.138\)</span>, and the
’effective number of independent reads is <span class="math inline">\((192+600+290)/2.75 \approx 393\)</span>. Then, to
estimate the probability of the genotype <span class="math inline">\(1\alpha:3\beta:2\delta\)</span>, for example, we
calculate</p>
<p><span class="math display">\[
\begin{align}
&amp;\textrm{Pr}\left(2A/D \approx 1.324\ \&amp;\&amp;\ 2B/D \approx
4.138\ |\ \textrm{true genotype is } 1\alpha:3\beta:2\delta\right)
\approx \\
&amp;\textrm{Pr}\left(2A/D \approx 1.324\ \&amp;\&amp;\ 2B/D \approx
4.138\ |\ 393 \textrm{ independent reads are simulated from a}\right. \\
&amp;\textrm{multinomial distribution with relative probabilities of }
1.06, 3.13 \textrm{ and } 2 \textrm{ for } \alpha, \beta, \\
&amp;\left.\textrm{ and } \delta \textrm{ reads respectively}\right)
\end{align}
\]</span> These are actually approximate likelihoods; operationally,
<span class="math inline">\(2A/D \approx 1.324\)</span> is the estimated
probability that the simulated <span class="math inline">\(2A/D\)</span>
is between <span class="math inline">\((0.99\times1.324)\)</span> and
<span class="math inline">\((1.01\times1.324)\)</span>. To calculate
posterior probabilities, we still need a prior estimate of the
population-level genotype frequencies. There is some element of
circularity here since we ultimately are estimating genotypes (which go
in to the population-level genotype frequency estimates). Ideally we
would have an independent way of assessing population frequencies
(e.g. widespread ddPCR), but in the absence of that we have obtained
crude prior estimates (see Table 1 below) by qualitative assessments of
figures like Figure 1. Continuing with our example (and assuming
European ancestry), we estimate</p>
<p><span class="math display">\[
\begin{align}
  \textrm{lik}\left(1\alpha:3\beta:2\delta\right) &amp;\sim
\textrm{Pr}\left(1\alpha:3\beta:2\delta\right)\ \times \\
  &amp;\; \; \; \; \; \textrm{Pr}\left(2A/D \approx 1.324\ \&amp;\&amp;\
2B/D \approx 4.138\ | \right. \\
  &amp;\; \; \; \; \; \; \; \; \; \; \; \left.\textrm{genotype} =
1\alpha:3\beta:2\delta\right) \\
  &amp;= 0.434 \times 0.0002521 \\
  \\
  \textrm{lik}\left(1\alpha:4\beta:2\delta\right) &amp;\sim
\textrm{Pr}\left(1\alpha:4\beta:2\delta\right)\ \times \\
  &amp;\; \; \; \; \; \textrm{Pr}\left(2A/D \approx 1.324\ \&amp;\&amp;\
2B/D \approx 4.138\ | \right. \\
  &amp;\; \; \; \; \; \; \; \; \; \; \; \left. \textrm{genotype} =
1\alpha:4\beta:2\delta\right) \\
  &amp;= 0.03 \times 0.0012967 \\
  \\
  \textrm{lik}\left(1\alpha:5\beta:2\delta\right) &amp;\sim
\textrm{Pr}\left(1\alpha:5\beta:2\delta\right)\ \times \\
  &amp;\; \; \; \; \; \textrm{Pr}\left(2A/D \approx 1.324\ \&amp;\&amp;\
2B/D \approx 4.138\ | \right. \\
  &amp;\; \; \; \; \; \; \; \; \; \; \; \left. \textrm{genotype} =
1\alpha:5\beta:2\delta\right) \\
  &amp;= 0.0004 \times 0.000019
\end{align}
\]</span></p>
<p>The posterior probability of a <span class="math inline">\(1\alpha:3\beta:2\delta\)</span> genotype is
estimated at</p>
<p><span class="math display">\[
\frac{0.434 \times 0.0002521}{0.434 \times 0.0002521 + 0.03 \times
0.0012967 + 0.0004 \times 0.000019} = 0.738
\]</span></p>
<p>Note that this estimate (and many others) is highly sensitive to the
(unknown) prior population frequencies. <span class="math inline">\(\textrm{Pr}\left(\textrm{data}\ |\
\textrm{genotype}\right)\)</span> is highest for <span class="math inline">\(1\alpha:4\beta:2\delta\)</span> by a factor of
<span class="math inline">\(5\)</span>, but this is swamped out by the
very low prior for the allele frequency.</p>
<p>Table 1. Prior estimates of tryptase genotype frequencies, stratified
by ancestry: EUR = European, EAS = East Asian, AFR = African, SAS =
South Asian</p>
<div class="lefttable">
<table class="table">
<thead><tr class="header">
<th align="left">
<span class="math inline">\(\alpha\)</span> copy
number</th>
<th align="left">
<span class="math inline">\(\beta\)</span> copy
number</th>
<th align="left">EUR</th>
<th align="left">EAS</th>
<th align="left">AFR</th>
<th align="left">SAS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">0</td>
<td align="left">2</td>
<td align="left">0.0001</td>
<td align="left">0.025</td>
<td align="left">0.001</td>
<td align="left">0.004</td>
</tr>
<tr class="even">
<td align="left">0</td>
<td align="left">3</td>
<td align="left">0.013</td>
<td align="left">0.037</td>
<td align="left">0.015</td>
<td align="left">0.053</td>
</tr>
<tr class="odd">
<td align="left">0</td>
<td align="left">4</td>
<td align="left">0.219</td>
<td align="left">0.032</td>
<td align="left">0.274</td>
<td align="left">0.165</td>
</tr>
<tr class="even">
<td align="left">0</td>
<td align="left">5</td>
<td align="left">0.020028</td>
<td align="left">0.004</td>
<td align="left">0.1</td>
<td align="left">0.0325</td>
</tr>
<tr class="odd">
<td align="left">0</td>
<td align="left">6</td>
<td align="left">0.0004</td>
<td align="left">0.0002</td>
<td align="left">0.008</td>
<td align="left">0.0006</td>
</tr>
<tr class="even">
<td align="left">0</td>
<td align="left">7</td>
<td align="left">0.00005</td>
<td align="left">0.00002</td>
<td align="left">0.0003</td>
<td align="left">0.00005</td>
</tr>
<tr class="odd">
<td align="left">0</td>
<td align="left">8</td>
<td align="left">0.00001</td>
<td align="left">0.000002</td>
<td align="left">0.0001</td>
<td align="left">0.000005</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">1</td>
<td align="left">0.0001</td>
<td align="left">0.0021</td>
<td align="left">0.0001</td>
<td align="left">0.0005</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">2</td>
<td align="left">0.013</td>
<td align="left">0.22</td>
<td align="left">0.012</td>
<td align="left">0.017</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">3</td>
<td align="left">0.434</td>
<td align="left">0.23</td>
<td align="left">0.39</td>
<td align="left">0.444</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">4</td>
<td align="left">0.03</td>
<td align="left">0.012</td>
<td align="left">0.06</td>
<td align="left">0.01</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">5</td>
<td align="left">0.0004</td>
<td align="left">0.0005</td>
<td align="left">0.005</td>
<td align="left">0.0005</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">6</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.0001</td>
<td align="left">0.00005</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">7</td>
<td align="left">0.000001</td>
<td align="left">0.000003</td>
<td align="left">0.00001</td>
<td align="left">0.000005</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">1</td>
<td align="left">0.001</td>
<td align="left">0.004</td>
<td align="left">0.0007</td>
<td align="left">0.0003</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">2</td>
<td align="left">0.208</td>
<td align="left">0.42</td>
<td align="left">0.126536</td>
<td align="left">0.256129</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">3</td>
<td align="left">0.03</td>
<td align="left">0.005</td>
<td align="left">0.003</td>
<td align="left">0.004</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">4</td>
<td align="left">0.004</td>
<td align="left">0.0005</td>
<td align="left">0.0001</td>
<td align="left">0.001</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">5</td>
<td align="left">0.0001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.0001</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">6</td>
<td align="left">0.00004</td>
<td align="left">0.000002</td>
<td align="left">0.000001</td>
<td align="left">0.00001</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">1</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">2</td>
<td align="left">0.018</td>
<td align="left">0.006</td>
<td align="left">0.003</td>
<td align="left">0.01</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">3</td>
<td align="left">0.005</td>
<td align="left">0.001</td>
<td align="left">0.0003</td>
<td align="left">0.0003</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">4</td>
<td align="left">0.0004</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">5</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.0001</td>
<td align="left">0.00001</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">2</td>
<td align="left">0.002</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0003</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">3</td>
<td align="left">0.0008</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">4</td>
<td align="left">0.00005</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">5</td>
<td align="left">0.00001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
<td align="left">0.0001</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">2</td>
<td align="left">0.0002</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">3</td>
<td align="left">0.0001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">4</td>
<td align="left">0.00004</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">5</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
<td align="left">0.00001</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">2</td>
<td align="left">0.00003</td>
<td align="left">0.000001</td>
<td align="left">0.000001</td>
<td align="left">0.00001</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">3</td>
<td align="left">0.00001</td>
<td align="left">0.000001</td>
<td align="left">0.000001</td>
<td align="left">0.000001</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">4</td>
<td align="left">0.000001</td>
<td align="left">0.000001</td>
<td align="left">0.000001</td>
<td align="left">0.000001</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="appendix">Appendix<a class="anchor" aria-label="anchor" href="#appendix"></a>
</h2>
<p>Tryptase region consensus alignments for different alleles, using the
numbering of Maun et al., 2019</p>
<p><mark class="tg">Green text</mark> = cDNA sequence is potentially
incorrect</p>
<p><mark class="bgr">Red highlight</mark> = fixed difference between
delta and (alpha + beta)</p>
<p><mark class="bgy">Yellow highlight</mark> = polymorphic within alpha
+ beta, and can be used to discriminate between alpha &amp;/or beta
alleles</p>
<p><mark class="bgx">Grey highlight</mark> = polymorphic within the
specific allele/gene</p>
<div class="section level4">
<h4 id="bases-370-449">Bases 370 – 449<a class="anchor" aria-label="anchor" href="#bases-370-449"></a>
</h4>
<div class="superwide">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">GCCCGTCCTG GCGAGCC<mark class="bgx">CG</mark>G
CCTACG<mark class="bgx">T</mark>GGC CCCTGGTGAG
TCCCAGCC<mark class="bgx">G</mark>G GGTCCACCCT GCCCCTCACC
ACATTCCACA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">GCCCGTCCTG GCGAGCC<mark class="bgx">GC</mark>G
CCTACGCGGC CCCTGGTGAG TCCCAGCCGG GGTCCACCCT GCCCCTCACC ACATTCCACA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">GCCCGTCCTG GCGAGCCGCG
CCTACG<mark class="bgx">C</mark>GGC CCCTGGTGAG
TCCCAGCC<mark class="bgx">G</mark>G GGTCCACCCT GCCCCTCACC
ACATTCCACA</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-450-549">Bases 450 – 549<a class="anchor" aria-label="anchor" href="#bases-450-549"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">GATCAGGGCC TGGGTGGGTT CTGGGGAGGC CGGGCTGGCC
CCCACACAGG GAAGGGCTGG <mark class="bgr">T</mark>CCCAGGCGT
GGGGC<mark class="bgr">G</mark>GCTT C<mark class="bgr">T</mark>TGGTCCTG
ACCTGGCACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">GGTCAGGGCC TGGGTGGGTT CTGGGGAGGC
<mark class="bgy">T</mark>GGGCTGGCC CCCACACAGG
GAAGGG<mark class="bgx">G</mark>TGG GCCCAGGC<mark class="bgy">G</mark>T
GGGGCTGCTT CCTGGTCCTG ACCTGGCACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">G<mark class="bgx">G</mark>TCAGGGCC TGGGTGGGTT
CTGGGGAGG<mark class="bgx">C</mark><mark class="bgy">C</mark>GGGCTGGCC
CCCACACAGG GAAGGGCTGG GCCCAGGC<mark class="bgx">G</mark>T GGGGCTGCTT
CCTGGTCCTG ACCTGGCACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono">G<mark class="bgx">G</mark>TCAGGGCC TGGGTGGGTT
CTGGGGAGG<mark class="bgx">C</mark><mark class="bgy">C</mark>GGGCTGGCC
CCCACACAGG GAAGGGCTGG GCCCAGGC<mark class="bgy">C</mark>T GGGGCTGCTT
CCTGGTCCTG ACCTGGCACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">G<mark class="bgx">G</mark>TCAGGGCC TGGGTGGGTT
CTGGGGAGG<mark class="bgx">C</mark><mark class="bgy">C</mark>GGGCTGGCC
CCCACACAGG GAAGGGCTGG GCCCAGGC<mark class="bgy">C</mark>T GGGGCTGCTT
CCTGGTCCTG ACCTGGCACC</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-550-649">Bases 550 – 649<a class="anchor" aria-label="anchor" href="#bases-550-649"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono"><mark class="tg">T</mark>GCCCC<mark class="tg">A</mark>GCC
CCAGGCCAGG CCCTGCAGCA
A<mark class="bgr">A</mark>CGGGCAT<mark class="bgr">T</mark>
GTTGGGGG<mark class="bgr">G</mark>C AG<mark class="bgx">G</mark>AGGCCCC
CAGGAGCAAG TGGCCCTGGC AGGTGAGCCT GAGAGTCCGC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono"><mark class="tg">T</mark>GCCCC<mark class="tg">A</mark>GCC
CCAG<mark class="bgy">T</mark>CCAGG CCCTGCAGC<mark class="bgy">A</mark>
AG<mark class="bgy">C</mark>GGG<mark class="bgy">T</mark>ATC
GT<mark class="bgy">C</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC
AGGTGAGCCT GAGAGTCC<mark class="bgy">G</mark>C</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">TGCCCCAGCC CCAG<mark class="bgy">G</mark>CCAGG
CCCTGCAGC<mark class="bgy">G</mark>
AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC
GT<mark class="bgy">C</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC
AGGTGAGCCT GAGAGTCC<mark class="bgy">A</mark>C</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono"><mark class="tg">T</mark>GCCCC<mark class="tg">A</mark>GCC
CCAG<mark class="bgy">G</mark>CCAGG CCCTGCAGC<mark class="bgy">G</mark>
AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC
GT<mark class="bgy">T</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC
AGGTGAGCCT GAGAGTCC<mark class="bgy">A</mark>C</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">TGCCCCAGCC CCAG<mark class="bgy">G</mark>CCAGG
CCCTGCAGC<mark class="bgy">G</mark>
AG<mark class="bgy">T</mark>GGG<mark class="bgy">C</mark>ATC
GT<mark class="bgy">T</mark>GGGGGTC AGGAGGCCCC CAGGAGCAAG TGGCCCTGGC
AGGTGAGCCT GAGAGTCC<mark class="bgy">G</mark>C</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-650-749">Bases 650 – 749<a class="anchor" aria-label="anchor" href="#bases-650-749"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">GGCCCATACT GGATGCACTT CTGCGGGGGC
TCC<mark class="bgx">C</mark>TCATCC ACCCCCAGTG
GGTGCT<mark class="bgr">A</mark>ACC GC<mark class="bgx">G</mark>GCGCACT
GC<mark class="bgx">G</mark>TGG<mark class="bgr">A</mark>ACC GTGAGTCTCC
<mark class="bgr">T</mark>GGGGCCTGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">G<mark class="bgy">A</mark>CC<mark class="bgy">G</mark>ATACT
GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC
GC<mark class="bgy">G</mark>GCGCACT GC<mark class="bgy">C</mark>TGGGACC
GTGAGTCTCC C<mark class="bgx">A</mark>GGGCCTGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">G<mark class="bgy">G</mark>CC<mark class="bgy">C</mark>ATACT
GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC
GC<mark class="bgy">A</mark>GCGCACT GC<mark class="bgy">G</mark>TGGGACC
GTGAGTCTCC CGGGGCCTGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono">G<mark class="bgy">G</mark>CC<mark class="bgy">C</mark>ATACT
GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC
GC<mark class="bgy">A</mark>GCGCACT GC<mark class="bgy">G</mark>TGGGACC
GTGAGTCTCC CGGGGCCTGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">G<mark class="bgy">A</mark>CC<mark class="bgy">G</mark>ATACT
GGATGCACTT CTGCGGGGGC TCCCTCATCC ACCCCCAGTG GGTGCTGACC
GC<mark class="bgx">A</mark>GCGCACT GC<mark class="bgx">G</mark>TGGGACC
GTGAGTCTCC CGGGGCCTGG</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-750-849">Bases 750 – 849<a class="anchor" aria-label="anchor" href="#bases-750-849"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">AGGGGTGGGC AAGGGCTGGA TGTGAGCCCT GGCTCCCGGG
TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG
GAC<mark class="bgr">A</mark>TCAAGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">AGGGGTGGG<mark class="bgy">C</mark> AAGGGCTGGA
TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC
GCTGCCCAGG GACGTCAAGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">AGGGGTGGG<mark class="bgy">G</mark> AAGGGCTGGA
TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC
GCTGCCCAGG GACGTCAAGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono">AGGGGTGGG<mark class="bgy">G</mark> AAGGGCTGGA
TGTGAGCCCT GGCTCCCGGG TGCTCCTGGG GGCTGCCCAG GGCCCTGAGT GGGATCCTCC
GCTGCCCAGG GACGTCAAGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">A<mark class="bgx">G</mark>GGGTGGG<mark class="bgy">G</mark>
AAGGGCTGGA TGTGAGCCCT GGCTCCCGG<mark class="bgx">G</mark> TGCTCCTGGG
GGCTGCCCAG GGCCCTGAGT GGGATCCTCC GCTGCCCAGG GACGTCAAGG</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-850-949">Bases 850 – 949<a class="anchor" aria-label="anchor" href="#bases-850-949"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">ATCTGGCCGC CCTCAGGGTG CAACTGCGGG AGCAGCACCT
CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC GTGCACCCAC AGTTCTACAT</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">ATCTGGCC<mark class="bgy">A</mark>C CCTCAGGGTG
CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC
TGCC<mark class="bgx">G</mark>GTCAG CAGGATCATC GTGCACCCAC
AGTTCTACA<mark class="bgy">T</mark></span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">ATCTGGCC<mark class="bgy">G</mark>C CCTCAGGGTG
CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC
GTGCACCCAC AGTTCTACA<mark class="bgy">C</mark></span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono">ATCTGGCC<mark class="bgy">G</mark>C CCTCAGGGTG
CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC
GTGCACCCAC AGTTCTACA<mark class="bgy">C</mark></span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">ATCTGGCC<mark class="bgy">G</mark>C CCTCAGGGTG
CAACTGCGGG AGCAGCACCT CTACTACCAG GACCAGCTGC TGCCGGTCAG CAGGATCATC
GTGCACCCAC AGTTCTACA<mark class="bgy">C</mark></span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-950-1049">Bases 950 – 1049<a class="anchor" aria-label="anchor" href="#bases-950-1049"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">CATCCAGACC GG<mark class="bgr">G</mark>GCGGACA
TCGCCCTGCT GGAGCTGGAG GAGCCCGTGA ACATCTCCAG
CCAC<mark class="bgr">A</mark>TCCAC AC<mark class="bgx">G</mark>GTCACGC
TGCCCCCTGC CTCGGAGACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">C<mark class="bgy">AT</mark>CCAGA<mark class="bgy">CT</mark>
GGAGCGGA<mark class="bgy">T</mark>A TCGCCCTGCT GGAGCTGGAG
GAGCC<mark class="bgy">C</mark>GTGA AC<mark class="bgy">A</mark>TCTCCAG
CC<mark class="bgy">G</mark>CGTCCAC ACGGTCA<mark class="bgy">TG</mark>C
TGCCCCCTGC CTC<mark class="bgy">G</mark>GAGACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">C<mark class="bgy">GC</mark>CCAGA<mark class="bgy">TC</mark>
GGAGCGGA<mark class="bgy">C</mark>A TCGCCCTGCT GGAGCTGGAG
GAGCC<mark class="bgy">G</mark>GTGA AC<mark class="bgy">G</mark>TCTCCAG
CC<mark class="bgy">A</mark>CGTCCAC ACGGTCA<mark class="bgy">CC</mark>C
TGCCCCCTGC CTC<mark class="bgy">A</mark>GAGACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono">C<mark class="bgy">GC</mark>CCAGA<mark class="bgy">TC</mark>
GGAGCGGA<mark class="bgy">C</mark>A TCGCCCTGCT GGAGCTGGAG
GAGCC<mark class="bgy">G</mark>GTGA
A<mark class="bgx">G</mark><mark class="bgy">G</mark>TCTCCAG
CC<mark class="bgy">A</mark>CGTCCAC ACGGTCA<mark class="bgy">CC</mark>C
TGCCCCCTGC CTC<mark class="bgy">A</mark>GAGACC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">C<mark class="bgy">GC</mark>CCAGA<mark class="bgy">TC</mark>
GGAGCGGA<mark class="bgy">C</mark>A TCGCCCTGCT GGAGCTGGAG
GAGCC<mark class="bgy">G</mark>GTGA AC<mark class="bgy">G</mark>TCTCCAG
CC<mark class="bgy">A</mark>CGTCCAC ACGGTCA<mark class="bgy">CC</mark>C
TGCCCCCTGC CTC<mark class="bgy">A</mark>GAGACC</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1050-1149">Bases 1050 – 1149<a class="anchor" aria-label="anchor" href="#bases-1050-1149"></a>
</h4>
<p><mark class="bgp">beta3FS location is colored pink</mark> [<span class="mono">–/C</span>]</p>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG
A<mark class="bgr">C</mark>GTGGACAA T<mark class="bgr">A</mark>ATGGTGGG
<mark class="bgr">GTGT</mark>TGGGGA CAG<mark class="bgr">C</mark>GGGAGG
<mark class="bgr">CC</mark>GGGCCAGG
<mark class="bgr">TGGGCACCAA</mark></span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG
ATGTGGACAA TGATGGTGGG ––TCTGGGGA CAGTGG<mark class="bgy">G</mark>AGG
TGGGGCCAGG GTCTTAGCCA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta1
</td>
<td>
<span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG
ATGTGGACAA TGATGGTGGG ––TCTGGGGA CAGTGG<mark class="bgy">–</mark>AGG
TGGGGCCAGG GTCTTAGCCA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta2
</td>
<td>
<span class="mono">TTCCCCCCGG GGATGCCGTG CTGGGTCACT GGCTGGGGCG
ATGTGGACAA TGATGGTGGG ––TCTGGGGA CAGTGG<mark class="bgy">–</mark>AGG
TGGGGCCAGG GTCTTAGCCA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta3
</td>
<td>
<span class="mono">TT<mark class="bgp">CCCCCC</mark>GG GGATGCCGTG
CTGGGTCACT GGCTGGGGCG ATGTGGACAA TGATGGTGGG ––TCTGGGGA
CAGTGG<mark class="bgy">–</mark>AGG TGGGGCCAGG GTCTTAGCCA</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1150-1249">Bases 1150 – 1249<a class="anchor" aria-label="anchor" href="#bases-1150-1249"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono"><mark class="bgr">GTCACAGCCA</mark><mark class="bgr">CAGGCCAGTC</mark><mark class="bgr">CGTGGGGTGA</mark>                                             <mark class="bgr">CCCAG</mark>C<mark class="bgr">C</mark>G<mark class="bgr">G</mark>C
CCC<mark class="bgr">AGAC</mark>C<mark class="bgr">C</mark>G
GCT<mark class="bgr">C</mark>CAC<mark class="bgr">G</mark>C<mark class="bgr">C</mark></span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">CAGCCCAGCC CCTGG––––– –––––––––– ––––––––––
–––––––––– ––––––GCTC CAGGTGGGGG TTGCCCGGCC
CCCTCCT<mark class="bgy">C</mark>AG GCTGCACCCT</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">CAGCCCAGCC CCTGGCCAGG GTCTTAGCCA CAGCCCAGCC
CCTGGGCTCC CTCTGGGCTC CAGGTGGGGG TTGCCCGGCC
CCCTCCT<mark class="bgy">G</mark>AG GCTGCACCCT</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1250-1349">Bases 1250 – 1349<a class="anchor" aria-label="anchor" href="#bases-1250-1349"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">C<mark class="bgr">CC</mark>C<mark class="bgr">T</mark>CC<mark class="bgr">G</mark>CC
<mark class="bgr">CC</mark>CAG<mark class="bgr">T</mark>GC<mark class="bgr">A</mark>C
CT<mark class="bgr">G</mark>CC<mark class="bgr">G</mark>CCGC
CAT<mark class="bgr">AC</mark>CC<mark class="bgr">G</mark>CT
GAAG<mark class="bgr">G</mark>AGGTG
<mark class="bgr">G</mark>A<mark class="bgr">A</mark>GTCCCC<mark class="bgr">G</mark>
TA<mark class="bgr">G</mark>TGGAAAA
CCAC<mark class="bgr">C</mark>TTTG<mark class="bgr">C</mark><mark class="bgr">A</mark>ACGC<mark class="bgr">GG</mark>AAT
A<mark class="bgr">T</mark>CAC<mark class="bgr">ACC</mark>GG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">CTTCCCCACC TGCAGAGC<mark class="bgy">C</mark>C
CTC<mark class="bgx">C</mark>CACCGC CATTTCC<mark class="bgy">C</mark>CT
GAAGCAGGTG AAGGTCCCCA TAATGGAAAA CCACATTTGT GACGCAAAAT ACCACCTTGG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">CTTCCCCACC TGCAGAGC<mark class="bgy">G</mark>C
CTCCCACCGC CATTTCC<mark class="bgy">T</mark>CT GAAGCAGGTG AAGGTCCCCA
TAATGGAAAA CCACATTTGT GACGCAAAAT ACCACCTTGG</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1350-1449">Bases 1350 – 1449<a class="anchor" aria-label="anchor" href="#bases-1350-1449"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">C<mark class="bgr">CT</mark>C<mark class="bgr">C</mark>A<mark class="bgr">T</mark>ACG
GG<mark class="bgr">CC</mark>AC<mark class="bgr">AG</mark>C<mark class="bgr">T</mark>
T<mark class="bgr">T</mark>C<mark class="bgr">AA</mark>ATCGT
CCG<mark class="bgr">C</mark>GA<mark class="bgr">T</mark>GAC ATGCTGTGTG
C<mark class="bgr">G</mark>GGGA<mark class="bgr">G</mark>CG<mark class="bgr">A</mark><mark class="bgr">AA</mark>A<mark class="bgr">TCA</mark>CGAC</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha1
</td>
<td>
<span class="mono">CGCCTACACG GGAGACGACG
TCCGCATC<mark class="bgy">A</mark>T CCGTGACGAC ATGCTGTGTG
CCGGGAACA<mark class="bgy">G</mark> CC<mark class="bgy">A</mark>GAGGGAC
TC<mark class="bgy">C</mark>TGC<mark class="bgy">A</mark>AGG TGGGCCCCGC
GTGTCCCCCG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha2
</td>
<td>
<span class="mono">CGCCTACACG GGAGACGACG
TCCGCATC<mark class="bgy">A</mark>T CCGTGACGAC ATGCTGTGTG
CCGGGAACA<mark class="bgy">C</mark> CC<mark class="bgy">G</mark>GAGGGAC
TC<mark class="bgy">A</mark>TGC<mark class="bgy">C</mark>AGG TGGGCCCCGC
GTGT<mark class="bgx">C</mark>CCCCG</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">CGCCTACACG GGAGACGACG
TCCGCATC<mark class="bgy">G</mark>T CCGTGACGAC ATGCTGTGTG CCGGGAACAC
CCGGAGGGAC TC<mark class="bgy">A</mark>TGC<mark class="bgy">C</mark>AGG
TGGGCCCCGC GTGTCCCCCG</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1450-1549">Bases 1450 – 1549<a class="anchor" aria-label="anchor" href="#bases-1450-1549"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono"><mark class="bgr">GAGCGCATCC</mark><mark class="bgr">CTCATCCTGA</mark><mark class="bgr">CCCCCGAAGC</mark><mark class="bgr">CTGGCCA</mark>GCG AGC<mark class="bgr">A</mark>CTGACC
TCTGACCTTC CCAGGG<mark class="bgr">T</mark>GAC TCTGGAGGGC
CCCTGGT<mark class="bgr">C</mark>TG CAAGGTGAAT</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha1
</td>
<td>
<span class="mono">CCCCCCGC<mark class="bgy">A</mark>C
CCC<mark class="bgx">A</mark>A<mark class="bgx">T</mark>CCCC
ACTCCC<mark class="bgy">G</mark>GG<mark class="bgy">T</mark>
CTG<mark class="bgy">G</mark>TCGG<mark class="bgy">T</mark>G AGCGCTGACC
TCTGACCTTC CCAGGGCGAC TC<mark class="bgy">T</mark>GGAGGGC CCCTGGTGTG
CAAGGTGAAT</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha2
</td>
<td>
<span class="mono">CCCCCCGC<mark class="bgy">A</mark>C
CCCAA<mark class="bgy">C</mark>CCCC
ACTCCC<mark class="bgy">A</mark>GG<mark class="bgy">C</mark>
CTG<mark class="bgy">T</mark>TCGG<mark class="bgy">C</mark>G AGCGCTGACC
TCTGACCTTC CCAGGGCGAC TC<mark class="bgy">T</mark>GGAGGGC CCCTGGTGTG
CAAGGTGAAT</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">CCCCCCGC<mark class="bgx">C</mark>C CCCAACCCCC
ACTCCCAGGC CTG<mark class="bgx">T</mark>TCGGCG AGCGCTGACC TCTGACCTTC
CCAGGGCGAC TC<mark class="bgy">C</mark>GGAGGGC CCCTGGTGTG
CAAGGTGAAT</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1550-1649">Bases 1550 – 1649<a class="anchor" aria-label="anchor" href="#bases-1550-1649"></a>
</h4>
<p>Alignment near putative functional alpha/beta mutation in
<mark class="bgg">green</mark> (cf. Maun et al., 2019 supplement)</p>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">GGCACCT<mark class="bgr">AA</mark>C TGCAGGCGGG
CGTGGTCAGC TGGGA<mark class="bgr">G</mark>GAG<mark class="bgr">A</mark>
GCTGTGCCCA GCCCAACCGG CCTGGCATCT ACACCCGTGT CACCTACTAC TTGGACTGGA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">GGCACCTGGC T<mark class="bgy">A</mark>CAGGCGGG
CGTGGTCAGC TGGG<mark class="bgg">A</mark>CGAGG GCTGTGCCCA GCCCAACCGG
CCTGGCATCT ACACCCGTGT CACCTACTAC TTGGACTGGA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">GGCACCTGGC T<mark class="bgy">G</mark>CAGGCGGG
CGTGGTCAGC TGGG<mark class="bgg">G</mark>CGAGG GCTGTGCCCA GCCCAACCGG
CCTGGCATCT ACACCCGTGT CACCTACTAC TTGGACTGGA</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level4">
<h4 id="bases-1650-1749">Bases 1650 – 1749<a class="anchor" aria-label="anchor" href="#bases-1650-1749"></a>
</h4>
<div class="evensuperwider">
<table class="table table table-condensed">
<tr style="border-top: 1px solid #ccc;">
<td width="8%;">
delta
</td>
<td>
<span class="mono">TCCACCACTA TGTCCCCAA<mark class="bgr">G</mark>
AAGCC<mark class="bgr">C</mark>TGAG <mark class="bgr">C</mark>CAGGCCTGG
GGTGTCCACC <mark class="bgr">C</mark>GGGTCACTG GAGGGCCAGC CCCTCCTGTC
CAAA<mark class="bgr">C</mark>CACCA CTGCTTCCTA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
alpha
</td>
<td>
<span class="mono">TCCACCACTA TGTCCCCAAA AAGCCGTGAG TCAGGCCTGG
GTTGGCCACC TGGGTCACTG
GAG<mark class="bgx">GA</mark>CCA<mark class="bgx">AC</mark>
CCCT<mark class="bgx">G</mark>CTGTC CAAAACACCA CTGCTTCCTA</span>
</td>
</tr>
<tr style="border-top: 1px solid #ccc;">
<td>
beta
</td>
<td>
<span class="mono">TCCACCACTA TGTCCCCAAA AAGCCGTGAG TCAGGCCTGG
G<mark class="bgx">T</mark>TG<mark class="bgx">G</mark>CCACC TGGGTCACTG
GAG<mark class="bgx">GA</mark>CCA<mark class="bgx">AC</mark>
CCCT<mark class="bgx">G</mark>CTGTC CAAAACACCA CTGCTTCCTA</span>
</td>
</tr>
</table>
</div>
</div>
<div class="section level3">
<h3 id="citing-tryplik">Citing <code>TrypLik</code><a class="anchor" aria-label="anchor" href="#citing-tryplik"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.19/TrypLik" class="external-link">TrypLik</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"TrypLik"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'TrypLik' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Carroll J, Wall J (2024). _TrypLik: Calculates Tryptase Variant</span></span>
<span><span class="co">#&gt;   Likelihoods_. R package version 0.1.0,</span></span>
<span><span class="co">#&gt;   &lt;https://hibio.github.io/TrypLik/&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {TrypLik: Calculates Tryptase Variant Likelihoods},</span></span>
<span><span class="co">#&gt;     author = {Jonathan Carroll and Jeff Wall},</span></span>
<span><span class="co">#&gt;     year = {2024},</span></span>
<span><span class="co">#&gt;     note = {R package version 0.1.0},</span></span>
<span><span class="co">#&gt;     url = {https://hibio.github.io/TrypLik/},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.19/TrypLik" class="external-link">TrypLik</a></em>
package Carroll et al., 2024 was made possible thanks to:</p>
<ul>
<li>R R Core Team, 2024</li>
<li>
<em><a href="https://bioconductor.org/packages/3.19/BiocStyle" class="external-link">BiocStyle</a></em>
Oleś, 2024</li>
<li>
<em><a href="https://bioconductor.org/packages/3.19/biocthis" class="external-link">biocthis</a></em>
Collado-Torres, 2024</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> Xie,
2024</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
McLean, 2017</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
Allaire et al., 2024</li>
</ul>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.19/BiocStyle" class="external-link">BiocStyle</a></em>
Oleś, 2024 with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> Xie, 2024
and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
Allaire et al., 2024 running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
McLean, 2017.</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.26. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-carroll2024tryplik"></a><a href="#cite-carroll2024tryplik">[2]</a><cite>
J. Carroll and J. Wall. <em>TrypLik: Calculates Tryptase Variant
Likelihoods</em>. R package version 0.1.0. 2024. URL:
<a href="https://hibio.github.io/TrypLik/">https://hibio.github.io/TrypLik/</a>.</cite>
</p>
<p>
<a id="bib-chollet2021novel"></a><a href="#cite-chollet2021novel">[3]</a><cite>
M. B. Chollet, A. Kovalszki, and C. Akin. “A novel case of
beta-tryptase/TPSB2 allele duplication”. In: <em>Journal of Allergy and
Clinical Immunology</em> 147(2) supplement (2021). DOI:
<a href="https://doi.org/10.1016/j.jaci.2020.12.491" class="external-link">10.1016/j.jaci.2020.12.491</a>.</cite>
</p>
<p>
<a id="bib-chovanec2022unpublished"></a><a href="#cite-chovanec2022unpublished">[4]</a><cite>
Chovanec. “Unpublished data”. TBA. 2022.</cite>
</p>
<p>
<a id="bib-colladotorres2024automate"></a><a href="#cite-colladotorres2024automate">[5]</a><cite>
L. Collado-Torres. <em>Automate package and project setup for
Bioconductor packages</em>. <a href="https://github.com/lcolladotor/biocthisbiocthis" class="external-link uri">https://github.com/lcolladotor/biocthisbiocthis</a> - R
package version 1.14.0. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.biocthis" class="external-link">10.18129/B9.bioc.biocthis</a>.
URL:
<a href="http://www.bioconductor.org/packages/biocthis" class="external-link">http://www.bioconductor.org/packages/biocthis</a>.</cite>
</p>
<p>
<a id="bib-glover2022hereditary"></a><a href="#cite-glover2022hereditary">[6]</a><cite>
S. C. Glover, A. Carlyle, and J. J. Lyons. “Hereditary alpha-tryptasemia
despite normal tryptase-encoding gene copy number owing to copy number
loss in trans”. In: <em>Annals of Allergy, Asthma &amp; Immunology</em>
128(4) (2022), pp. 460-461. DOI:
<a href="https://doi.org/10.1016/j.anai.2021.12.008" class="external-link">10.1016/j.anai.2021.12.008</a>.</cite>
</p>
<p>
<a id="bib-lyonsetal2016elevated"></a><a href="#cite-lyonsetal2016elevated">[7]</a><cite>
J. J. Lyons et al. “Elevated basal serum tryptase identifies a
multisystem disorder associated with increased TPSAB1 copy number.” In:
<em>Nat Genet.</em> 48(12) (2016), pp. 1564-1569. DOI:
<a href="https://doi.org/doi:10.1038/ng.3696" class="external-link">doi:10.1038/ng.3696</a>.</cite>
</p>
<p>
<a id="bib-maunetal2019allosteric"></a><a href="#cite-maunetal2019allosteric">[8]</a><cite>
H. R. Maun et al. “An Allosteric Anti-tryptase Antibody for the
Treatment of Mast Cell-Mediated Severe Asthma”. In: <em>Cell</em> 179(2)
(2019), pp. 417-431. DOI:
<a href="https://doi.org/doi:10.1016/j.cell.2019.09.009" class="external-link">doi:10.1016/j.cell.2019.09.009</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[9]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2024biocstyle"></a><a href="#cite-ole2024biocstyle">[10]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.32.0. 2024. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2024language"></a><a href="#cite-2024language">[11]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2024. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-xie2024knitr"></a><a href="#cite-xie2024knitr">[12]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.46. 2024. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.5.1    TrypLik_0.1.0    RefManageR_1.4.0 BiocStyle_2.32.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] sass_0.4.9          utf8_1.2.4          generics_0.1.3     </span></span>
<span><span class="co">#&gt;  [4] xml2_1.3.6          stringi_1.8.3       digest_0.6.35      </span></span>
<span><span class="co">#&gt;  [7] magrittr_2.0.3      evaluate_0.23       grid_4.4.0         </span></span>
<span><span class="co">#&gt; [10] timechange_0.3.0    bookdown_0.39       fastmap_1.1.1      </span></span>
<span><span class="co">#&gt; [13] plyr_1.8.9          jsonlite_1.8.8      backports_1.4.1    </span></span>
<span><span class="co">#&gt; [16] BiocManager_1.30.22 httr_1.4.7          purrr_1.0.2        </span></span>
<span><span class="co">#&gt; [19] fansi_1.0.6         scales_1.3.0        bibtex_0.5.1       </span></span>
<span><span class="co">#&gt; [22] textshaping_0.3.7   jquerylib_0.1.4     cli_3.6.2          </span></span>
<span><span class="co">#&gt; [25] rlang_1.1.3         munsell_0.5.1       withr_3.0.0        </span></span>
<span><span class="co">#&gt; [28] cachem_1.0.8        yaml_2.3.8          tools_4.4.0        </span></span>
<span><span class="co">#&gt; [31] memoise_2.0.1       colorspace_2.1-0    vctrs_0.6.5        </span></span>
<span><span class="co">#&gt; [34] R6_2.5.1            lifecycle_1.0.4     lubridate_1.9.3    </span></span>
<span><span class="co">#&gt; [37] stringr_1.5.1       fs_1.6.4            ragg_1.3.0         </span></span>
<span><span class="co">#&gt; [40] pkgconfig_2.0.3     desc_1.4.3          pkgdown_2.0.9      </span></span>
<span><span class="co">#&gt; [43] pillar_1.9.0        bslib_0.7.0         gtable_0.3.5       </span></span>
<span><span class="co">#&gt; [46] glue_1.7.0          Rcpp_1.0.12         systemfonts_1.0.6  </span></span>
<span><span class="co">#&gt; [49] highr_0.10          xfun_0.43           tibble_3.2.1       </span></span>
<span><span class="co">#&gt; [52] knitr_1.46          farver_2.1.1        htmltools_0.5.8.1  </span></span>
<span><span class="co">#&gt; [55] rmarkdown_2.26      labeling_0.4.3      compiler_4.4.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jonathan Carroll, Jeff Wall, Human Immunology Biosciences (HI·Bio).</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
